{
 "metadata": {
  "name": "",
  "signature": "sha256:d0e830071bde8a0d9804f79e945dedb509e813d3592191b2598361f84debeb84"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "MYC analyses in Pathways"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "In Leiserson et al. MYC is left out, though it is presented as a rather \"hot\" node with frequent copynumber amplifications.<p>\n",
      "Leiserson:<i>Cancer cells are thought to harbor multiple driver mutations that\n",
      "perturb multiple biological functions15. Consistent with this model,\n",
      "we found that four pairs of subnetworks, including TP53 and NOTCH\n",
      "signaling, TP53 and RTK signaling, PI3K signaling and the cohesin\n",
      "complex, and PI3K signaling and the ASCOM complex, exhibited\n",
      "significant co-occurrence (P < 0.05, multiple-hypotheses corrected)\n",
      "across the pan-cancer cohort (Fig. 2b) or in individual cancer types\n",
      "(Fig. 2c). Multiple pairs of genes within these subnetworks showed\n",
      "co-occurring mutations (Supplementary Table 24). In contrast,\n",
      "mutually exclusive mutations are typically expected within a pathway\n",
      "and not across pathways32,33.</i>\n",
      "\n",
      "<p>Ref 32\tYeang, C.-H., McCormick, F. & Levine, A. Combinatorial patterns of somatic gene\n",
      "mutations in cancer. FASEB J. 22, 2605\u20132622 (2008).\n",
      "<p>Ref 33\tVandin, F., Upfal, E. & Raphael, B.J. De novo discovery of mutated driver pathways\n",
      "in cancer. Genome Res. 22, 375\u2013385 (2012).<p>\n",
      "Where does MYC belong to?\n",
      "\n"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import NotebookImport\n",
      "from TCGA_analysis_BRCA_import import *\n",
      "from IPython.display import HTML"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "importing IPython notebook from TCGA_analysis_BRCA_import.ipynb\n",
        "Loading Focal Amplification data"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "Loading RNASeq data"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "Loading RNASeq data of controls"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "Loading CNV Data"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "Loading Somatic Mutation Data"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-A8-A07C-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-BH-A0HN-01A not found\n",
        "TCGA-BH-A0B8-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-BH-A0HF-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-A7-A4SC-01A not found\n",
        "TCGA-AR-A0TU-01A not found\n",
        "TCGA-AR-A1AT-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-E2-A1LS-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-A2-A0CZ-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-BH-A0HL-01A not found\n",
        "TCGA-AN-A0G0-01A not found\n",
        "TCGA-BH-A0B1-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-B6-A0I8-01A not found"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "TCGA-B6-A0I6-01A not found\n",
        "Loading Clinical Data"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "Samples: 1086"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "  --Focal Data: 1086\n",
        "  --CNV:        1086\n",
        "  --RNASeq:     1081\n",
        "  --Somatic:    968\n",
        "  --Clinical:   1058\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which pathways are affected in MYC focally amplified samples"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "For all pathways described by Leiserson: Check which pathway is more often co-mutated by MYC"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "TP53_pathway=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\", \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\"]\n",
      "STAG1_pathway=[\"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\"]\n",
      "MLL_pathway=[\"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\"]\n",
      "BAP1_pathway=[\"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\"]\n",
      "KRAS_pathway=[\"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\"]\n",
      "HLA_pathway=[\"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\"]\n",
      "HER2_pathway=[\"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\"]\n",
      "CDON_pathway=[\"BOC\", \"CDON\"]\n",
      "KEAP1_pathway=[\"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\"]\n",
      "SPOP_pathway=[\"MYD88\", \"SPOP\"]\n",
      "RUNX1_pathway=[\"CBFB\", \"RUNX1\",\"ELF4\"]\n",
      "NOTCH_pathway=[\"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\"]\n",
      "ARID_pathway=[\"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\"]\n",
      "SMARC_pathway=[\"SMARCB1\", \"SMARCA4\"]\n",
      "SMC_pathway=[\"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\"]\n",
      "CLASP_pathway=[\"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "pathways={\"TP53_pathway\": TP53_pathway, \"STAG1_pathway\":STAG1_pathway, \"MLL_pathway\":MLL_pathway,\"BAP1_pathway\":BAP1_pathway,\"KRAS_pathway\":KRAS_pathway, \"HLA_pathway\":HLA_pathway,  \"HER2_pathway\":HER2_pathway, \"CDON_pathway\":CDON_pathway, \"KEAP1_pathway\":KEAP1_pathway,\"SPOP_pathway\":SPOP_pathway, \"RUNX1_pathway\":RUNX1_pathway, \"NOTCH_pathway\":NOTCH_pathway, \"ARID_pathway\":ARID_pathway, \"SMARC_pathway\":SMARC_pathway, \"SMC_pathway\":SMC_pathway,\"CLASP_pathway\":CLASP_pathway }\n",
      "\n",
      "MYC_pos_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_pos_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAG1_pathway</th><td>2</td><td>59</td><td>54</td><td>853</td><td>0.54</td><td>0.57</td><td>9.15</td><td>0.54</td><td>0.88</td><td>14.13</td><td>0.54</td><td>0.30</td><td>4.77</td></tr><tr><th>KRAS_pathway</th><td>21</td><td>40</td><td>373</td><td>534</td><td>0.75</td><td>0.35</td><td>5.55</td><td>0.75</td><td>0.88</td><td>14.06</td><td>0.75</td><td>0.19</td><td>2.97</td></tr><tr><th>CLASP_pathway</th><td>4</td><td>57</td><td>21</td><td>886</td><td>2.96</td><td>0.07</td><td>1.07</td><td>2.96</td><td>0.07</td><td>1.07</td><td>2.96</td><td>0.98</td><td>15.73</td></tr><tr><th>SMARC_pathway</th><td>0</td><td>61</td><td>9</td><td>898</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.56</td><td>8.88</td></tr><tr><th>SPOP_pathway</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.72</td><td>11.55</td></tr><tr><th>RUNX1_pathway</th><td>1</td><td>60</td><td>58</td><td>849</td><td>0.24</td><td>0.17</td><td>2.72</td><td>0.24</td><td>0.98</td><td>15.70</td><td>0.24</td><td>0.10</td><td>1.59</td></tr><tr><th>MLL_pathway</th><td>14</td><td>47</td><td>115</td><td>792</td><td>2.05</td><td>0.03086</td><td>0.49383</td><td>2.05</td><td>0.02357</td><td>0.37711</td><td>2.05</td><td>0.99</td><td>15.84</td></tr><tr><th>TP53_pathway</th><td>49</td><td>12</td><td>544</td><td>363</td><td>2.72</td><td>0.00159</td><td>0.02544</td><td>2.72</td><td>0.00085</td><td>0.01354</td><td>2.72</td><td>1.00</td><td>16.00</td></tr><tr><th>ARID_pathway</th><td>3</td><td>58</td><td>66</td><td>841</td><td>0.66</td><td>0.80</td><td>12.72</td><td>0.66</td><td>0.83</td><td>13.25</td><td>0.66</td><td>0.35</td><td>5.63</td></tr><tr><th>KEAP1_pathway</th><td>4</td><td>57</td><td>30</td><td>877</td><td>2.05</td><td>0.16</td><td>2.57</td><td>2.05</td><td>0.16</td><td>2.57</td><td>2.05</td><td>0.94</td><td>15.09</td></tr><tr><th>SMC_pathway</th><td>2</td><td>59</td><td>27</td><td>880</td><td>1.10</td><td>0.70</td><td>11.26</td><td>1.10</td><td>0.56</td><td>8.91</td><td>1.10</td><td>0.73</td><td>11.61</td></tr><tr><th>NOTCH_pathway</th><td>12</td><td>49</td><td>130</td><td>777</td><td>1.46</td><td>0.26</td><td>4.19</td><td>1.46</td><td>0.17</td><td>2.70</td><td>1.46</td><td>0.90</td><td>14.47</td></tr><tr><th>BAP1_pathway</th><td>5</td><td>56</td><td>58</td><td>849</td><td>1.31</td><td>0.59</td><td>9.40</td><td>1.31</td><td>0.36</td><td>5.83</td><td>1.31</td><td>0.80</td><td>12.82</td></tr><tr><th>CDON_pathway</th><td>2</td><td>59</td><td>16</td><td>891</td><td>1.89</td><td>0.32</td><td>5.04</td><td>1.89</td><td>0.32</td><td>5.04</td><td>1.89</td><td>0.90</td><td>14.42</td></tr><tr><th>HLA_pathway</th><td>1</td><td>60</td><td>27</td><td>880</td><td>0.54</td><td>1.00</td><td>16.00</td><td>0.54</td><td>0.84</td><td>13.48</td><td>0.54</td><td>0.46</td><td>7.40</td></tr><tr><th>HER2_pathway</th><td>18</td><td>43</td><td>167</td><td>740</td><td>1.85</td><td>0.04239</td><td>0.67832</td><td>1.85</td><td>0.02890</td><td>0.46235</td><td>1.85</td><td>0.99</td><td>15.78</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 2,
       "text": [
        "<IPython.core.display.HTML at 0x369fa90>"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Remove TP53 from TP53 pathway"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "TP53_pathway=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\", \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\"]\n",
      "STAG1_pathway=[\"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\"]\n",
      "MLL_pathway=[\"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\"]\n",
      "BAP1_pathway=[\"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\"]\n",
      "KRAS_pathway=[\"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\"]\n",
      "HLA_pathway=[\"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\"]\n",
      "HER2_pathway=[\"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\"]\n",
      "CDON_pathway=[\"BOC\", \"CDON\"]\n",
      "KEAP1_pathway=[\"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\"]\n",
      "SPOP_pathway=[\"MYD88\", \"SPOP\"]\n",
      "RUNX1_pathway=[\"CBFB\", \"RUNX1\",\"ELF4\"]\n",
      "NOTCH_pathway=[\"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\"]\n",
      "ARID_pathway=[\"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\"]\n",
      "SMARC_pathway=[\"SMARCB1\", \"SMARCA4\"]\n",
      "SMC_pathway=[\"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\"]\n",
      "CLASP_pathway=[\"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "pathways={\"TP53_pathway\": TP53_pathway, \"STAG1_pathway\":STAG1_pathway, \"MLL_pathway\":MLL_pathway,\"BAP1_pathway\":BAP1_pathway,\"KRAS_pathway\":KRAS_pathway, \"HLA_pathway\":HLA_pathway,  \"HER2_pathway\":HER2_pathway, \"CDON_pathway\":CDON_pathway, \"KEAP1_pathway\":KEAP1_pathway,\"SPOP_pathway\":SPOP_pathway, \"RUNX1_pathway\":RUNX1_pathway, \"NOTCH_pathway\":NOTCH_pathway, \"ARID_pathway\":ARID_pathway, \"SMARC_pathway\":SMARC_pathway, \"SMC_pathway\":SMC_pathway,\"CLASP_pathway\":CLASP_pathway }\n",
      "\n",
      "MYC_pos_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_pos_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAG1_pathway</th><td>2</td><td>59</td><td>54</td><td>853</td><td>0.54</td><td>0.57</td><td>9.15</td><td>0.54</td><td>0.88</td><td>14.13</td><td>0.54</td><td>0.30</td><td>4.77</td></tr><tr><th>KRAS_pathway</th><td>21</td><td>40</td><td>373</td><td>534</td><td>0.75</td><td>0.35</td><td>5.55</td><td>0.75</td><td>0.88</td><td>14.06</td><td>0.75</td><td>0.19</td><td>2.97</td></tr><tr><th>CLASP_pathway</th><td>4</td><td>57</td><td>21</td><td>886</td><td>2.96</td><td>0.07</td><td>1.07</td><td>2.96</td><td>0.07</td><td>1.07</td><td>2.96</td><td>0.98</td><td>15.73</td></tr><tr><th>SMARC_pathway</th><td>0</td><td>61</td><td>9</td><td>898</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.56</td><td>8.88</td></tr><tr><th>SPOP_pathway</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.72</td><td>11.55</td></tr><tr><th>RUNX1_pathway</th><td>1</td><td>60</td><td>58</td><td>849</td><td>0.24</td><td>0.17</td><td>2.72</td><td>0.24</td><td>0.98</td><td>15.70</td><td>0.24</td><td>0.10</td><td>1.59</td></tr><tr><th>MLL_pathway</th><td>14</td><td>47</td><td>115</td><td>792</td><td>2.05</td><td>0.03086</td><td>0.49383</td><td>2.05</td><td>0.02357</td><td>0.37711</td><td>2.05</td><td>0.99</td><td>15.84</td></tr><tr><th>TP53_pathway</th><td>34</td><td>27</td><td>401</td><td>506</td><td>1.59</td><td>0.09</td><td>1.36</td><td>1.59</td><td>0.05</td><td>0.85</td><td>1.59</td><td>0.97</td><td>15.52</td></tr><tr><th>ARID_pathway</th><td>3</td><td>58</td><td>66</td><td>841</td><td>0.66</td><td>0.80</td><td>12.72</td><td>0.66</td><td>0.83</td><td>13.25</td><td>0.66</td><td>0.35</td><td>5.63</td></tr><tr><th>KEAP1_pathway</th><td>4</td><td>57</td><td>30</td><td>877</td><td>2.05</td><td>0.16</td><td>2.57</td><td>2.05</td><td>0.16</td><td>2.57</td><td>2.05</td><td>0.94</td><td>15.09</td></tr><tr><th>SMC_pathway</th><td>2</td><td>59</td><td>27</td><td>880</td><td>1.10</td><td>0.70</td><td>11.26</td><td>1.10</td><td>0.56</td><td>8.91</td><td>1.10</td><td>0.73</td><td>11.61</td></tr><tr><th>NOTCH_pathway</th><td>12</td><td>49</td><td>130</td><td>777</td><td>1.46</td><td>0.26</td><td>4.19</td><td>1.46</td><td>0.17</td><td>2.70</td><td>1.46</td><td>0.90</td><td>14.47</td></tr><tr><th>BAP1_pathway</th><td>5</td><td>56</td><td>58</td><td>849</td><td>1.31</td><td>0.59</td><td>9.40</td><td>1.31</td><td>0.36</td><td>5.83</td><td>1.31</td><td>0.80</td><td>12.82</td></tr><tr><th>CDON_pathway</th><td>2</td><td>59</td><td>16</td><td>891</td><td>1.89</td><td>0.32</td><td>5.04</td><td>1.89</td><td>0.32</td><td>5.04</td><td>1.89</td><td>0.90</td><td>14.42</td></tr><tr><th>HLA_pathway</th><td>1</td><td>60</td><td>27</td><td>880</td><td>0.54</td><td>1.00</td><td>16.00</td><td>0.54</td><td>0.84</td><td>13.48</td><td>0.54</td><td>0.46</td><td>7.40</td></tr><tr><th>HER2_pathway</th><td>18</td><td>43</td><td>167</td><td>740</td><td>1.85</td><td>0.04239</td><td>0.67832</td><td>1.85</td><td>0.02890</td><td>0.46235</td><td>1.85</td><td>0.99</td><td>15.78</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 3,
       "text": [
        "<IPython.core.display.HTML at 0x619bad0>"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which pathways are affected in MYC CN>4 samples"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "For all pathways described by Leiserson: Check which pathway is more often co-mutated by MYC"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "TP53_pathway=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\", \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\"]\n",
      "STAG1_pathway=[\"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\"]\n",
      "MLL_pathway=[\"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\"]\n",
      "BAP1_pathway=[\"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\"]\n",
      "KRAS_pathway=[\"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\"]\n",
      "HLA_pathway=[\"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\"]\n",
      "HER2_pathway=[\"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\"]\n",
      "CDON_pathway=[\"BOC\", \"CDON\"]\n",
      "KEAP1_pathway=[\"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\"]\n",
      "SPOP_pathway=[\"MYD88\", \"SPOP\"]\n",
      "RUNX1_pathway=[\"CBFB\", \"RUNX1\",\"ELF4\"]\n",
      "NOTCH_pathway=[\"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\"]\n",
      "ARID_pathway=[\"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\"]\n",
      "SMARC_pathway=[\"SMARCB1\", \"SMARCA4\"]\n",
      "SMC_pathway=[\"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\"]\n",
      "CLASP_pathway=[\"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "pathways={\"TP53_pathway\": TP53_pathway, \"STAG1_pathway\":STAG1_pathway, \"MLL_pathway\":MLL_pathway,\"BAP1_pathway\":BAP1_pathway,\"KRAS_pathway\":KRAS_pathway, \"HLA_pathway\":HLA_pathway,  \"HER2_pathway\":HER2_pathway, \"CDON_pathway\":CDON_pathway, \"KEAP1_pathway\":KEAP1_pathway,\"SPOP_pathway\":SPOP_pathway, \"RUNX1_pathway\":RUNX1_pathway, \"NOTCH_pathway\":NOTCH_pathway, \"ARID_pathway\":ARID_pathway, \"SMARC_pathway\":SMARC_pathway, \"SMC_pathway\":SMC_pathway,\"CLASP_pathway\":CLASP_pathway }\n",
      "\n",
      "MYC_pos_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_pos_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    copynumber = \"n/a\"\n",
      "    log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "    if (log2 != None):\n",
      "        copynumber = 2.0 * 2.0 ** log2\n",
      "    else:\n",
      "        copynumber=\"n/a\"\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if copynumber != \"n/a\" and pathway_affected and copynumber > 4:\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and  copynumber > 4:\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and pathway_affected and not  copynumber > 4:\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and not  copynumber > 4:\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAG1_pathway</th><td>1</td><td>111</td><td>55</td><td>800</td><td>0.13</td><td>0.01558</td><td>0.24926</td><td>0.13</td><td>1.00</td><td>15.99</td><td>0.13</td><td>0.00721</td><td>0.11543</td></tr><tr><th>KRAS_pathway</th><td>42</td><td>70</td><td>351</td><td>504</td><td>0.86</td><td>0.54</td><td>8.63</td><td>0.86</td><td>0.79</td><td>12.70</td><td>0.86</td><td>0.27</td><td>4.31</td></tr><tr><th>CLASP_pathway</th><td>4</td><td>108</td><td>21</td><td>834</td><td>1.47</td><td>0.52</td><td>8.32</td><td>1.47</td><td>0.33</td><td>5.23</td><td>1.47</td><td>0.85</td><td>13.55</td></tr><tr><th>SMARC_pathway</th><td>1</td><td>111</td><td>8</td><td>847</td><td>0.95</td><td>1.00</td><td>16.00</td><td>0.95</td><td>0.67</td><td>10.74</td><td>0.95</td><td>0.72</td><td>11.52</td></tr><tr><th>SPOP_pathway</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.54</td><td>8.63</td></tr><tr><th>RUNX1_pathway</th><td>2</td><td>110</td><td>57</td><td>798</td><td>0.25</td><td>0.05</td><td>0.88</td><td>0.25</td><td>0.99</td><td>15.92</td><td>0.25</td><td>0.02352</td><td>0.37631</td></tr><tr><th>MLL_pathway</th><td>16</td><td>96</td><td>113</td><td>742</td><td>1.09</td><td>0.77</td><td>12.28</td><td>1.09</td><td>0.42</td><td>6.78</td><td>1.09</td><td>0.69</td><td>10.96</td></tr><tr><th>TP53_pathway</th><td>90</td><td>22</td><td>502</td><td>353</td><td>2.88</td><td>0.00001</td><td>0.00011</td><td>2.88</td><td>0.00000</td><td>0.00006</td><td>2.88</td><td>1.00</td><td>16.00</td></tr><tr><th>ARID_pathway</th><td>8</td><td>104</td><td>61</td><td>794</td><td>1.00</td><td>1.00</td><td>16.00</td><td>1.00</td><td>0.56</td><td>8.95</td><td>1.00</td><td>0.59</td><td>9.52</td></tr><tr><th>KEAP1_pathway</th><td>5</td><td>107</td><td>29</td><td>826</td><td>1.33</td><td>0.58</td><td>9.31</td><td>1.33</td><td>0.36</td><td>5.72</td><td>1.33</td><td>0.81</td><td>12.94</td></tr><tr><th>SMC_pathway</th><td>5</td><td>107</td><td>24</td><td>831</td><td>1.62</td><td>0.37</td><td>5.91</td><td>1.62</td><td>0.24</td><td>3.81</td><td>1.62</td><td>0.89</td><td>14.27</td></tr><tr><th>NOTCH_pathway</th><td>19</td><td>93</td><td>123</td><td>732</td><td>1.22</td><td>0.48</td><td>7.65</td><td>1.22</td><td>0.27</td><td>4.39</td><td>1.22</td><td>0.81</td><td>12.94</td></tr><tr><th>BAP1_pathway</th><td>8</td><td>104</td><td>55</td><td>800</td><td>1.12</td><td>0.69</td><td>11.01</td><td>1.12</td><td>0.45</td><td>7.18</td><td>1.12</td><td>0.70</td><td>11.20</td></tr><tr><th>CDON_pathway</th><td>3</td><td>109</td><td>15</td><td>840</td><td>1.54</td><td>0.45</td><td>7.27</td><td>1.54</td><td>0.35</td><td>5.56</td><td>1.54</td><td>0.85</td><td>13.67</td></tr><tr><th>HLA_pathway</th><td>3</td><td>109</td><td>25</td><td>830</td><td>0.91</td><td>1.00</td><td>16.00</td><td>0.91</td><td>0.65</td><td>10.37</td><td>0.91</td><td>0.59</td><td>9.43</td></tr><tr><th>HER2_pathway</th><td>29</td><td>83</td><td>156</td><td>699</td><td>1.57</td><td>0.06</td><td>0.90</td><td>1.57</td><td>0.03849</td><td>0.61581</td><td>1.57</td><td>0.98</td><td>15.64</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 4,
       "text": [
        "<IPython.core.display.HTML at 0xbd1111d0>"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Remove TP53 from TP53 pathway"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "TP53_pathway=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\", \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\"]\n",
      "STAG1_pathway=[\"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\"]\n",
      "MLL_pathway=[\"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\"]\n",
      "BAP1_pathway=[\"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\"]\n",
      "KRAS_pathway=[\"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\"]\n",
      "HLA_pathway=[\"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\"]\n",
      "HER2_pathway=[\"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\"]\n",
      "CDON_pathway=[\"BOC\", \"CDON\"]\n",
      "KEAP1_pathway=[\"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\"]\n",
      "SPOP_pathway=[\"MYD88\", \"SPOP\"]\n",
      "RUNX1_pathway=[\"CBFB\", \"RUNX1\",\"ELF4\"]\n",
      "NOTCH_pathway=[\"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\"]\n",
      "ARID_pathway=[\"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\"]\n",
      "SMARC_pathway=[\"SMARCB1\", \"SMARCA4\"]\n",
      "SMC_pathway=[\"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\"]\n",
      "CLASP_pathway=[\"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "pathways={\"TP53_pathway\": TP53_pathway, \"STAG1_pathway\":STAG1_pathway, \"MLL_pathway\":MLL_pathway,\"BAP1_pathway\":BAP1_pathway,\"KRAS_pathway\":KRAS_pathway, \"HLA_pathway\":HLA_pathway,  \"HER2_pathway\":HER2_pathway, \"CDON_pathway\":CDON_pathway, \"KEAP1_pathway\":KEAP1_pathway,\"SPOP_pathway\":SPOP_pathway, \"RUNX1_pathway\":RUNX1_pathway, \"NOTCH_pathway\":NOTCH_pathway, \"ARID_pathway\":ARID_pathway, \"SMARC_pathway\":SMARC_pathway, \"SMC_pathway\":SMC_pathway,\"CLASP_pathway\":CLASP_pathway }\n",
      "\n",
      "MYC_pos_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_pos_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    copynumber = \"n/a\"\n",
      "    log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "    if (log2 != None):\n",
      "        copynumber = 2.0 * 2.0 ** log2\n",
      "    else:\n",
      "        copynumber=\"n/a\"\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if copynumber != \"n/a\" and pathway_affected and copynumber > 4:\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and  copynumber > 4:\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and pathway_affected and not  copynumber > 4:\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and not  copynumber > 4:\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAG1_pathway</th><td>1</td><td>111</td><td>55</td><td>800</td><td>0.13</td><td>0.01558</td><td>0.24926</td><td>0.13</td><td>1.00</td><td>15.99</td><td>0.13</td><td>0.00721</td><td>0.11543</td></tr><tr><th>KRAS_pathway</th><td>42</td><td>70</td><td>351</td><td>504</td><td>0.86</td><td>0.54</td><td>8.63</td><td>0.86</td><td>0.79</td><td>12.70</td><td>0.86</td><td>0.27</td><td>4.31</td></tr><tr><th>CLASP_pathway</th><td>4</td><td>108</td><td>21</td><td>834</td><td>1.47</td><td>0.52</td><td>8.32</td><td>1.47</td><td>0.33</td><td>5.23</td><td>1.47</td><td>0.85</td><td>13.55</td></tr><tr><th>SMARC_pathway</th><td>1</td><td>111</td><td>8</td><td>847</td><td>0.95</td><td>1.00</td><td>16.00</td><td>0.95</td><td>0.67</td><td>10.74</td><td>0.95</td><td>0.72</td><td>11.52</td></tr><tr><th>SPOP_pathway</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.54</td><td>8.63</td></tr><tr><th>RUNX1_pathway</th><td>2</td><td>110</td><td>57</td><td>798</td><td>0.25</td><td>0.05</td><td>0.88</td><td>0.25</td><td>0.99</td><td>15.92</td><td>0.25</td><td>0.02352</td><td>0.37631</td></tr><tr><th>MLL_pathway</th><td>16</td><td>96</td><td>113</td><td>742</td><td>1.09</td><td>0.77</td><td>12.28</td><td>1.09</td><td>0.42</td><td>6.78</td><td>1.09</td><td>0.69</td><td>10.96</td></tr><tr><th>TP53_pathway</th><td>59</td><td>53</td><td>375</td><td>480</td><td>1.42</td><td>0.09</td><td>1.38</td><td>1.42</td><td>0.04841</td><td>0.77451</td><td>1.42</td><td>0.97</td><td>15.50</td></tr><tr><th>ARID_pathway</th><td>8</td><td>104</td><td>61</td><td>794</td><td>1.00</td><td>1.00</td><td>16.00</td><td>1.00</td><td>0.56</td><td>8.95</td><td>1.00</td><td>0.59</td><td>9.52</td></tr><tr><th>KEAP1_pathway</th><td>5</td><td>107</td><td>29</td><td>826</td><td>1.33</td><td>0.58</td><td>9.31</td><td>1.33</td><td>0.36</td><td>5.72</td><td>1.33</td><td>0.81</td><td>12.94</td></tr><tr><th>SMC_pathway</th><td>5</td><td>107</td><td>24</td><td>831</td><td>1.62</td><td>0.37</td><td>5.91</td><td>1.62</td><td>0.24</td><td>3.81</td><td>1.62</td><td>0.89</td><td>14.27</td></tr><tr><th>NOTCH_pathway</th><td>19</td><td>93</td><td>123</td><td>732</td><td>1.22</td><td>0.48</td><td>7.65</td><td>1.22</td><td>0.27</td><td>4.39</td><td>1.22</td><td>0.81</td><td>12.94</td></tr><tr><th>BAP1_pathway</th><td>8</td><td>104</td><td>55</td><td>800</td><td>1.12</td><td>0.69</td><td>11.01</td><td>1.12</td><td>0.45</td><td>7.18</td><td>1.12</td><td>0.70</td><td>11.20</td></tr><tr><th>CDON_pathway</th><td>3</td><td>109</td><td>15</td><td>840</td><td>1.54</td><td>0.45</td><td>7.27</td><td>1.54</td><td>0.35</td><td>5.56</td><td>1.54</td><td>0.85</td><td>13.67</td></tr><tr><th>HLA_pathway</th><td>3</td><td>109</td><td>25</td><td>830</td><td>0.91</td><td>1.00</td><td>16.00</td><td>0.91</td><td>0.65</td><td>10.37</td><td>0.91</td><td>0.59</td><td>9.43</td></tr><tr><th>HER2_pathway</th><td>29</td><td>83</td><td>156</td><td>699</td><td>1.57</td><td>0.06</td><td>0.90</td><td>1.57</td><td>0.03849</td><td>0.61581</td><td>1.57</td><td>0.98</td><td>15.64</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 5,
       "text": [
        "<IPython.core.display.HTML at 0x369f6d0>"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which genes are affected in MYC focally amplified samples"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Check all genes which appear in any pathway in Leiserson paper\n",
      "Only output genes where any p-value is lower than 0.05"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\",\n",
      "       \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\",\n",
      "       \"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\",\n",
      "       \"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\",\n",
      "       \"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\",\n",
      "       \"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\",\n",
      "       \"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\",\n",
      "       \"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\",\n",
      "       \"BOC\", \"CDON\",\n",
      "       \"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\",\n",
      "       \"MYD88\", \"SPOP\",\n",
      "       \"CBFB\", \"RUNX1\",\"ELF4\",\n",
      "       \"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\",\n",
      "       \"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\",\n",
      "       \"SMARCB1\", \"SMARCA4\",\n",
      "       \"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\",\n",
      "       \"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "gene_count = 0\n",
      "for gene in genes:\n",
      "    gene_count += 1\n",
      "    MYC_pos_gene_affected = 0\n",
      "    MYC_pos_gene_unaffected = 0\n",
      "    MYC_neg_gene_affected = 0\n",
      "    MYC_neg_gene_unaffected = 0\n",
      "\n",
      "    for sample in samples.values():\n",
      "        if not sample.somatic_mutation_data:\n",
      "            continue\n",
      "        if sample.checkFocalGeneAmp(gene) or gene in sample.genes_affected:\n",
      "            if sample.checkFocalGeneAmp(\"MYC\"):\n",
      "                MYC_pos_gene_affected += 1\n",
      "            else:\n",
      "                MYC_neg_gene_affected += 1\n",
      "        else:\n",
      "            if sample.checkFocalGeneAmp(\"MYC\"):\n",
      "                MYC_pos_gene_unaffected += 1\n",
      "            else:\n",
      "                MYC_neg_gene_unaffected += 1\n",
      "\n",
      "    oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]])\n",
      "    oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='greater')\n",
      "    oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='less')\n",
      "\n",
      "        \n",
      "    s += \"<tr><th>\"+gene+\"</th><td>\"+str(MYC_pos_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_pos_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "    if pvalue < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "    if pvalue_greater < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "    if pvalue_less < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "    else: \n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "print \"Genes: \"+str(gene_count)\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Genes: 134\n"
       ]
      },
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>CCND1</th><td>10</td><td>51</td><td>182</td><td>725</td><td>0.78</td><td>0.62</td><td>82.90</td><td>0.78</td><td>0.80</td><td>107.69</td><td>0.78</td><td>0.31</td><td>40.95</td></tr><tr><th>CDKN2A</th><td>1</td><td>60</td><td>4</td><td>903</td><td>3.76</td><td>0.28</td><td>37.29</td><td>3.76</td><td>0.28</td><td>37.29</td><td>3.76</td><td>0.97</td><td>129.37</td></tr><tr><th>CUL9</th><td>1</td><td>60</td><td>8</td><td>899</td><td>1.87</td><td>0.44</td><td>59.60</td><td>1.87</td><td>0.44</td><td>59.60</td><td>1.87</td><td>0.89</td><td>119.84</td></tr><tr><th>NPM1</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>PTEN</th><td>2</td><td>59</td><td>33</td><td>874</td><td>0.90</td><td>1.00</td><td>134.00</td><td>0.90</td><td>0.66</td><td>88.65</td><td>0.90</td><td>0.62</td><td>82.92</td></tr><tr><th>TP53</th><td>36</td><td>25</td><td>261</td><td>646</td><td>3.56</td><td>0.00000</td><td>0.00032</td><td>3.56</td><td>0.00000</td><td>0.00027</td><td>3.56</td><td>1.00</td><td>134.00</td></tr><tr><th>AGBL2</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>ALS2CR8</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>AMOTL1</th><td>0</td><td>61</td><td>10</td><td>897</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.52</td><td>69.67</td></tr><tr><th>ANKRD12</th><td>0</td><td>61</td><td>20</td><td>887</td><td>0.00</td><td>0.63</td><td>84.58</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.27</td><td>35.97</td></tr><tr><th>BRPF1</th><td>0</td><td>61</td><td>6</td><td>901</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.68</td><td>90.58</td></tr><tr><th>CACHD1</th><td>1</td><td>60</td><td>6</td><td>901</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.93</td><td>125.04</td></tr><tr><th>CARNS1</th><td>2</td><td>59</td><td>44</td><td>863</td><td>0.66</td><td>0.76</td><td>102.23</td><td>0.66</td><td>0.80</td><td>107.49</td><td>0.66</td><td>0.43</td><td>58.24</td></tr><tr><th>CDKN2AIP</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>CELSR3</th><td>0</td><td>61</td><td>12</td><td>895</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.46</td><td>61.08</td></tr><tr><th>CHD8</th><td>0</td><td>61</td><td>15</td><td>892</td><td>0.00</td><td>0.62</td><td>82.66</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.37</td><td>50.11</td></tr><tr><th>EPHA3</th><td>3</td><td>58</td><td>22</td><td>885</td><td>2.08</td><td>0.20</td><td>27.46</td><td>2.08</td><td>0.20</td><td>27.46</td><td>2.08</td><td>0.93</td><td>125.08</td></tr><tr><th>HECW2</th><td>2</td><td>59</td><td>13</td><td>894</td><td>2.33</td><td>0.24</td><td>32.56</td><td>2.33</td><td>0.24</td><td>32.56</td><td>2.33</td><td>0.94</td><td>125.56</td></tr><tr><th>IFT140</th><td>2</td><td>59</td><td>7</td><td>900</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.98</td><td>131.96</td></tr><tr><th>IWS1</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>MAGI2</th><td>0</td><td>61</td><td>9</td><td>898</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.56</td><td>74.40</td></tr><tr><th>MAST3</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>MDM4</th><td>3</td><td>58</td><td>23</td><td>884</td><td>1.99</td><td>0.22</td><td>29.72</td><td>1.99</td><td>0.22</td><td>29.72</td><td>1.99</td><td>0.92</td><td>123.94</td></tr><tr><th>MLL5</th><td>2</td><td>59</td><td>10</td><td>897</td><td>3.04</td><td>0.17</td><td>23.02</td><td>3.04</td><td>0.17</td><td>23.02</td><td>3.04</td><td>0.97</td><td>129.33</td></tr><tr><th>PLEKHA8</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>PRDM2</th><td>1</td><td>60</td><td>7</td><td>900</td><td>2.14</td><td>0.41</td><td>54.55</td><td>2.14</td><td>0.41</td><td>54.55</td><td>2.14</td><td>0.91</td><td>122.53</td></tr><tr><th>PRKRIR</th><td>7</td><td>54</td><td>85</td><td>822</td><td>1.25</td><td>0.51</td><td>67.77</td><td>1.25</td><td>0.36</td><td>47.93</td><td>1.25</td><td>0.79</td><td>105.25</td></tr><tr><th>SCAPER</th><td>0</td><td>61</td><td>10</td><td>897</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.52</td><td>69.67</td></tr><tr><th>SETD2</th><td>2</td><td>59</td><td>15</td><td>892</td><td>2.02</td><td>0.29</td><td>39.02</td><td>2.02</td><td>0.29</td><td>39.02</td><td>2.02</td><td>0.91</td><td>122.47</td></tr><tr><th>SMG1</th><td>1</td><td>60</td><td>18</td><td>889</td><td>0.82</td><td>1.00</td><td>134.00</td><td>0.82</td><td>0.71</td><td>95.56</td><td>0.82</td><td>0.66</td><td>88.55</td></tr><tr><th>SMG5</th><td>0</td><td>61</td><td>8</td><td>899</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.59</td><td>79.45</td></tr><tr><th>SMG7</th><td>0</td><td>61</td><td>8</td><td>899</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.59</td><td>79.45</td></tr><tr><th>SNRK</th><td>2</td><td>59</td><td>5</td><td>902</td><td>6.12</td><td>0.07</td><td>8.96</td><td>6.12</td><td>0.07</td><td>8.96</td><td>6.12</td><td>0.99</td><td>133.07</td></tr><tr><th>SPTBN2</th><td>2</td><td>59</td><td>41</td><td>866</td><td>0.72</td><td>1.00</td><td>134.00</td><td>0.72</td><td>0.77</td><td>103.17</td><td>0.72</td><td>0.48</td><td>64.57</td></tr><tr><th>STK11IP</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>STRADA</th><td>0</td><td>61</td><td>7</td><td>900</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.63</td><td>84.84</td></tr><tr><th>SWAP70</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>TEP1</th><td>2</td><td>59</td><td>15</td><td>892</td><td>2.02</td><td>0.29</td><td>39.02</td><td>2.02</td><td>0.29</td><td>39.02</td><td>2.02</td><td>0.91</td><td>122.47</td></tr><tr><th>TRIP12</th><td>0</td><td>61</td><td>8</td><td>899</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.59</td><td>79.45</td></tr><tr><th>TTLL5</th><td>0</td><td>61</td><td>10</td><td>897</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.52</td><td>69.67</td></tr><tr><th>UACA</th><td>1</td><td>60</td><td>6</td><td>901</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.93</td><td>125.04</td></tr><tr><th>ZFP91</th><td>0</td><td>61</td><td>10</td><td>897</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.52</td><td>69.67</td></tr><tr><th>ZMIZ1</th><td>4</td><td>57</td><td>22</td><td>885</td><td>2.82</td><td>0.08</td><td>10.06</td><td>2.82</td><td>0.08</td><td>10.06</td><td>2.82</td><td>0.98</td><td>131.34</td></tr><tr><th>ZNF227</th><td>1</td><td>60</td><td>1</td><td>906</td><td>15.10</td><td>0.12</td><td>16.36</td><td>15.10</td><td>0.12</td><td>16.36</td><td>15.10</td><td>1.00</td><td>133.48</td></tr><tr><th>ZNF668</th><td>1</td><td>60</td><td>4</td><td>903</td><td>3.76</td><td>0.28</td><td>37.29</td><td>3.76</td><td>0.28</td><td>37.29</td><td>3.76</td><td>0.97</td><td>129.37</td></tr><tr><th>PDS5B</th><td>0</td><td>61</td><td>13</td><td>894</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.43</td><td>57.18</td></tr><tr><th>STAG1</th><td>1</td><td>60</td><td>11</td><td>896</td><td>1.36</td><td>0.54</td><td>72.92</td><td>1.36</td><td>0.54</td><td>72.92</td><td>1.36</td><td>0.83</td><td>110.98</td></tr><tr><th>STAG2</th><td>0</td><td>61</td><td>13</td><td>894</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.43</td><td>57.18</td></tr><tr><th>WAPAL</th><td>0</td><td>61</td><td>2</td><td>905</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.88</td><td>117.64</td></tr><tr><th>PDS5A</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>RAD21</th><td>1</td><td>60</td><td>5</td><td>902</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.95</td><td>127.34</td></tr><tr><th>SMC1A</th><td>0</td><td>61</td><td>11</td><td>896</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.49</td><td>65.24</td></tr><tr><th>KDM6A</th><td>2</td><td>59</td><td>14</td><td>893</td><td>2.16</td><td>0.27</td><td>35.79</td><td>2.16</td><td>0.27</td><td>35.79</td><td>2.16</td><td>0.93</td><td>124.07</td></tr><tr><th>MLL2</th><td>1</td><td>60</td><td>22</td><td>885</td><td>0.67</td><td>1.00</td><td>134.00</td><td>0.67</td><td>0.78</td><td>104.54</td><td>0.67</td><td>0.57</td><td>76.15</td></tr><tr><th>MLL3</th><td>7</td><td>54</td><td>60</td><td>847</td><td>1.83</td><td>0.18</td><td>24.70</td><td>1.83</td><td>0.12</td><td>16.17</td><td>1.83</td><td>0.95</td><td>126.96</td></tr><tr><th>E2F3</th><td>4</td><td>57</td><td>18</td><td>889</td><td>3.47</td><td>0.04435</td><td>5.94300</td><td>3.47</td><td>0.04435</td><td>5.94300</td><td>3.47</td><td>0.99</td><td>132.71</td></tr><tr><th>N4BP2</th><td>0</td><td>61</td><td>12</td><td>895</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.46</td><td>61.08</td></tr><tr><th>PROSER1</th><td>0</td><td>61</td><td>7</td><td>900</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.63</td><td>84.84</td></tr><tr><th>ASXL1</th><td>1</td><td>60</td><td>5</td><td>902</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.95</td><td>127.34</td></tr><tr><th>ASXL2</th><td>0</td><td>61</td><td>16</td><td>891</td><td>0.00</td><td>0.62</td><td>82.69</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.35</td><td>46.90</td></tr><tr><th>BAP1</th><td>0</td><td>61</td><td>7</td><td>900</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.63</td><td>84.84</td></tr><tr><th>ANKRD17</th><td>2</td><td>59</td><td>22</td><td>885</td><td>1.36</td><td>0.66</td><td>88.37</td><td>1.36</td><td>0.45</td><td>60.81</td><td>1.36</td><td>0.81</td><td>108.77</td></tr><tr><th>FOXK1</th><td>0</td><td>61</td><td>1</td><td>906</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.94</td><td>125.56</td></tr><tr><th>FOXK2</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>KDM1B</th><td>2</td><td>59</td><td>7</td><td>900</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.98</td><td>131.96</td></tr><tr><th>BRAF</th><td>1</td><td>60</td><td>12</td><td>895</td><td>1.24</td><td>0.57</td><td>76.82</td><td>1.24</td><td>0.57</td><td>76.82</td><td>1.24</td><td>0.80</td><td>107.84</td></tr><tr><th>KRAS</th><td>2</td><td>59</td><td>31</td><td>876</td><td>0.96</td><td>1.00</td><td>134.00</td><td>0.96</td><td>0.63</td><td>84.28</td><td>0.96</td><td>0.65</td><td>87.70</td></tr><tr><th>PIK3CA</th><td>14</td><td>47</td><td>317</td><td>590</td><td>0.55</td><td>0.07</td><td>9.26</td><td>0.55</td><td>0.98</td><td>131.65</td><td>0.55</td><td>0.03528</td><td>4.72707</td></tr><tr><th>ARHGAP4</th><td>0</td><td>61</td><td>6</td><td>901</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.68</td><td>90.58</td></tr><tr><th>EIF2AK1</th><td>1</td><td>60</td><td>6</td><td>901</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.93</td><td>125.04</td></tr><tr><th>FAM118B</th><td>2</td><td>59</td><td>6</td><td>901</td><td>5.09</td><td>0.09</td><td>11.47</td><td>5.09</td><td>0.09</td><td>11.47</td><td>5.09</td><td>0.99</td><td>132.58</td></tr><tr><th>NRAS</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>PLCE1</th><td>1</td><td>60</td><td>15</td><td>892</td><td>0.99</td><td>1.00</td><td>134.00</td><td>0.99</td><td>0.65</td><td>87.10</td><td>0.99</td><td>0.73</td><td>98.21</td></tr><tr><th>RASGRF2</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>RASGRP2</th><td>1</td><td>60</td><td>6</td><td>901</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.93</td><td>125.04</td></tr><tr><th>RGL1</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>VARS2</th><td>0</td><td>61</td><td>1</td><td>906</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.94</td><td>125.56</td></tr><tr><th>B2M</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>CD1D</th><td>1</td><td>60</td><td>5</td><td>902</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.95</td><td>127.34</td></tr><tr><th>HLA-A</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>HLA-B</th><td>0</td><td>61</td><td>8</td><td>899</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.59</td><td>79.45</td></tr><tr><th>P4HTM</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>EGFR</th><td>0</td><td>61</td><td>24</td><td>883</td><td>0.00</td><td>0.39</td><td>52.79</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.21</td><td>27.56</td></tr><tr><th>ERBB2</th><td>15</td><td>46</td><td>120</td><td>787</td><td>2.14</td><td>0.02042</td><td>2.73654</td><td>2.14</td><td>0.01532</td><td>2.05318</td><td>2.14</td><td>0.99</td><td>133.15</td></tr><tr><th>AREG</th><td>2</td><td>59</td><td>10</td><td>897</td><td>3.04</td><td>0.17</td><td>23.02</td><td>3.04</td><td>0.17</td><td>23.02</td><td>3.04</td><td>0.97</td><td>129.33</td></tr><tr><th>ELF3</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>ERBB4</th><td>2</td><td>59</td><td>21</td><td>886</td><td>1.43</td><td>0.65</td><td>87.30</td><td>1.43</td><td>0.43</td><td>57.85</td><td>1.43</td><td>0.83</td><td>110.94</td></tr><tr><th>LRIG1</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>OSMR</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>BOC</th><td>0</td><td>61</td><td>10</td><td>897</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.52</td><td>69.67</td></tr><tr><th>CDON</th><td>2</td><td>59</td><td>7</td><td>900</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.98</td><td>131.96</td></tr><tr><th>KEAP1</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>NFE2L2</th><td>1</td><td>60</td><td>5</td><td>902</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.32</td><td>43.42</td><td>3.01</td><td>0.95</td><td>127.34</td></tr><tr><th>CHD6</th><td>2</td><td>59</td><td>18</td><td>889</td><td>1.67</td><td>0.36</td><td>48.61</td><td>1.67</td><td>0.36</td><td>48.61</td><td>1.67</td><td>0.87</td><td>117.07</td></tr><tr><th>PTMA</th><td>0</td><td>61</td><td>1</td><td>906</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.94</td><td>125.56</td></tr><tr><th>WAC</th><td>1</td><td>60</td><td>8</td><td>899</td><td>1.87</td><td>0.44</td><td>59.60</td><td>1.87</td><td>0.44</td><td>59.60</td><td>1.87</td><td>0.89</td><td>119.84</td></tr><tr><th>MYD88</th><td>0</td><td>61</td><td>1</td><td>906</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.94</td><td>125.56</td></tr><tr><th>SPOP</th><td>0</td><td>61</td><td>4</td><td>903</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.77</td><td>103.24</td></tr><tr><th>CBFB</th><td>0</td><td>61</td><td>23</td><td>884</td><td>0.00</td><td>0.39</td><td>52.52</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.22</td><td>29.46</td></tr><tr><th>RUNX1</th><td>1</td><td>60</td><td>31</td><td>876</td><td>0.47</td><td>0.72</td><td>95.92</td><td>0.47</td><td>0.88</td><td>117.89</td><td>0.47</td><td>0.39</td><td>52.02</td></tr><tr><th>ELF4</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>FBXW7</th><td>2</td><td>59</td><td>15</td><td>892</td><td>2.02</td><td>0.29</td><td>39.02</td><td>2.02</td><td>0.29</td><td>39.02</td><td>2.02</td><td>0.91</td><td>122.47</td></tr><tr><th>NOTCH1</th><td>1</td><td>60</td><td>11</td><td>896</td><td>1.36</td><td>0.54</td><td>72.92</td><td>1.36</td><td>0.54</td><td>72.92</td><td>1.36</td><td>0.83</td><td>110.98</td></tr><tr><th>CCNE1</th><td>8</td><td>53</td><td>37</td><td>870</td><td>3.55</td><td>0.00528</td><td>0.70689</td><td>3.55</td><td>0.00528</td><td>0.70689</td><td>3.55</td><td>1.00</td><td>133.83</td></tr><tr><th>DCLRE1C</th><td>1</td><td>60</td><td>7</td><td>900</td><td>2.14</td><td>0.41</td><td>54.55</td><td>2.14</td><td>0.41</td><td>54.55</td><td>2.14</td><td>0.91</td><td>122.53</td></tr><tr><th>DLL1</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>DTX4</th><td>0</td><td>61</td><td>8</td><td>899</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.59</td><td>79.45</td></tr><tr><th>FAM101A</th><td>1</td><td>60</td><td>4</td><td>903</td><td>3.76</td><td>0.28</td><td>37.29</td><td>3.76</td><td>0.28</td><td>37.29</td><td>3.76</td><td>0.97</td><td>129.37</td></tr><tr><th>FOXM1</th><td>1</td><td>60</td><td>6</td><td>901</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.93</td><td>125.04</td></tr><tr><th>JAG1</th><td>0</td><td>61</td><td>13</td><td>894</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.43</td><td>57.18</td></tr><tr><th>KLF5</th><td>0</td><td>61</td><td>0</td><td>907</td><td>nan</td><td>1.00</td><td>134.00</td><td>nan</td><td>1.00</td><td>134.00</td><td>nan</td><td>1.00</td><td>134.00</td></tr><tr><th>LFNG</th><td>0</td><td>61</td><td>1</td><td>906</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.94</td><td>125.56</td></tr><tr><th>LPP</th><td>0</td><td>61</td><td>6</td><td>901</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.68</td><td>90.58</td></tr><tr><th>MAML1</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>MMS22L</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>NOTCH3</th><td>0</td><td>61</td><td>14</td><td>893</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.40</td><td>53.53</td></tr><tr><th>NOTCH4</th><td>0</td><td>61</td><td>10</td><td>897</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.52</td><td>69.67</td></tr><tr><th>SHPRH</th><td>0</td><td>61</td><td>11</td><td>896</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.49</td><td>65.24</td></tr><tr><th>ARID1A</th><td>0</td><td>61</td><td>27</td><td>880</td><td>0.00</td><td>0.41</td><td>54.58</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.17</td><td>22.55</td></tr><tr><th>PBRM1</th><td>1</td><td>60</td><td>6</td><td>901</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.37</td><td>49.16</td><td>2.50</td><td>0.93</td><td>125.04</td></tr><tr><th>ADNP</th><td>1</td><td>60</td><td>14</td><td>893</td><td>1.06</td><td>1.00</td><td>134.00</td><td>1.06</td><td>0.63</td><td>83.89</td><td>1.06</td><td>0.76</td><td>101.44</td></tr><tr><th>ARID1B</th><td>1</td><td>60</td><td>15</td><td>892</td><td>0.99</td><td>1.00</td><td>134.00</td><td>0.99</td><td>0.65</td><td>87.10</td><td>0.99</td><td>0.73</td><td>98.21</td></tr><tr><th>ARID2</th><td>0</td><td>61</td><td>11</td><td>896</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.49</td><td>65.24</td></tr><tr><th>SMARCB1</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>SMARCA4</th><td>0</td><td>61</td><td>7</td><td>900</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.63</td><td>84.84</td></tr><tr><th>NCAPD3</th><td>0</td><td>61</td><td>11</td><td>896</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.49</td><td>65.24</td></tr><tr><th>SMC4</th><td>0</td><td>61</td><td>8</td><td>899</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.59</td><td>79.45</td></tr><tr><th>NCAPG2</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>NCAPH2</th><td>0</td><td>61</td><td>3</td><td>904</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.82</td><td>110.21</td></tr><tr><th>SMC2</th><td>0</td><td>61</td><td>5</td><td>902</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.72</td><td>96.71</td></tr><tr><th>NCAPD2</th><td>2</td><td>59</td><td>3</td><td>904</td><td>10.21</td><td>0.03454</td><td>4.62781</td><td>10.21</td><td>0.03454</td><td>4.62781</td><td>10.21</td><td>1.00</td><td>133.71</td></tr><tr><th>CLASP1</th><td>1</td><td>60</td><td>8</td><td>899</td><td>1.87</td><td>0.44</td><td>59.60</td><td>1.87</td><td>0.44</td><td>59.60</td><td>1.87</td><td>0.89</td><td>119.84</td></tr><tr><th>CLASP2</th><td>1</td><td>60</td><td>7</td><td>900</td><td>2.14</td><td>0.41</td><td>54.55</td><td>2.14</td><td>0.41</td><td>54.55</td><td>2.14</td><td>0.91</td><td>122.53</td></tr><tr><th>CLIP2</th><td>2</td><td>59</td><td>7</td><td>900</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.11</td><td>14.16</td><td>4.36</td><td>0.98</td><td>131.96</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 6,
       "text": [
        "<IPython.core.display.HTML at 0xc4707bd0>"
       ]
      }
     ],
     "prompt_number": 6
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which genes are affected in MYC Copynumber > 4 samples"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\",\n",
      "       \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\",\n",
      "       \"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\",\n",
      "       \"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\",\n",
      "       \"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\",\n",
      "       \"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\",\n",
      "       \"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\",\n",
      "       \"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\",\n",
      "       \"BOC\", \"CDON\",\n",
      "       \"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\",\n",
      "       \"MYD88\", \"SPOP\",\n",
      "       \"CBFB\", \"RUNX1\",\"ELF4\",\n",
      "       \"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\",\n",
      "       \"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\",\n",
      "       \"SMARCB1\", \"SMARCA4\",\n",
      "       \"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\",\n",
      "       \"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "gene_count = 0\n",
      "for gene in genes:\n",
      "    gene_count += 1\n",
      "    MYC_pos_gene_affected = 0\n",
      "    MYC_pos_gene_unaffected = 0\n",
      "    MYC_neg_gene_affected = 0\n",
      "    MYC_neg_gene_unaffected = 0\n",
      "\n",
      "    for sample in samples.values():\n",
      "        if not sample.somatic_mutation_data:\n",
      "            continue\n",
      "        copynumber = \"n/a\"\n",
      "        log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "        if (log2 != None):\n",
      "            copynumber = 2.0 * 2.0 ** log2\n",
      "        else:\n",
      "            copynumber=\"n/a\"\n",
      "        if sample.checkFocalGeneAmp(gene) or gene in sample.genes_affected:\n",
      "            if copynumber != \"n/a\" and copynumber > 4:\n",
      "                MYC_pos_gene_affected += 1\n",
      "            elif  copynumber != \"n/a\" and not copynumber > 4:\n",
      "                MYC_neg_gene_affected += 1\n",
      "        else:\n",
      "            if copynumber != \"n/a\" and copynumber > 4:\n",
      "                MYC_pos_gene_unaffected += 1\n",
      "            elif  copynumber != \"n/a\" and not copynumber > 4:\n",
      "                MYC_neg_gene_unaffected += 1\n",
      "\n",
      "    oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]])\n",
      "    oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='greater')\n",
      "    oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='less')\n",
      "\n",
      "    #if (pvalue > 0.05 and pvalue_greater > 0.05 and pvalue_less > 0.05):\n",
      "        #continue\n",
      "        \n",
      "    s += \"<tr><th>\"+gene+\"</th><td>\"+str(MYC_pos_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_pos_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "    if pvalue < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "    if pvalue_greater < 0.0005: \n",
      "       s += \"<td>\"+\"{:.10f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.10f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    elif pvalue_greater < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "    if pvalue_less < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "    else: \n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "print \"Genes: \"+str(gene_count)\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Genes: 134\n"
       ]
      },
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>CCND1</th><td>27</td><td>85</td><td>165</td><td>690</td><td>1.33</td><td>0.26</td><td>34.35</td><td>1.33</td><td>0.14</td><td>19.02</td><td>1.33</td><td>0.91</td><td>121.33</td></tr><tr><th>CDKN2A</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>CUL9</th><td>0</td><td>112</td><td>9</td><td>846</td><td>0.00</td><td>0.61</td><td>81.59</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.33</td><td>44.04</td></tr><tr><th>NPM1</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>PTEN</th><td>3</td><td>109</td><td>32</td><td>823</td><td>0.71</td><td>0.79</td><td>105.69</td><td>0.71</td><td>0.79</td><td>106.22</td><td>0.71</td><td>0.41</td><td>54.53</td></tr><tr><th>TP53</th><td>63</td><td>49</td><td>233</td><td>622</td><td>3.43</td><td>0.00000</td><td>0.00000</td><td>3.43</td><td>0.0000000017</td><td>0.0000002249</td><td>3.43</td><td>1.00</td><td>134.00</td></tr><tr><th>AGBL2</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>ALS2CR8</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>AMOTL1</th><td>2</td><td>110</td><td>8</td><td>847</td><td>1.93</td><td>0.33</td><td>43.62</td><td>1.93</td><td>0.33</td><td>43.62</td><td>1.93</td><td>0.90</td><td>120.74</td></tr><tr><th>ANKRD12</th><td>0</td><td>112</td><td>20</td><td>835</td><td>0.00</td><td>0.15</td><td>20.66</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.08</td><td>11.13</td></tr><tr><th>BRPF1</th><td>1</td><td>111</td><td>5</td><td>850</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.85</td><td>114.41</td></tr><tr><th>CACHD1</th><td>1</td><td>111</td><td>6</td><td>849</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.81</td><td>108.57</td></tr><tr><th>CARNS1</th><td>5</td><td>107</td><td>41</td><td>814</td><td>0.93</td><td>1.00</td><td>134.00</td><td>0.93</td><td>0.63</td><td>84.90</td><td>0.93</td><td>0.55</td><td>74.28</td></tr><tr><th>CDKN2AIP</th><td>2</td><td>110</td><td>1</td><td>854</td><td>15.53</td><td>0.03689</td><td>4.94355</td><td>15.53</td><td>0.03689</td><td>4.94355</td><td>15.53</td><td>1.00</td><td>133.80</td></tr><tr><th>CELSR3</th><td>1</td><td>111</td><td>11</td><td>844</td><td>0.69</td><td>1.00</td><td>134.00</td><td>0.69</td><td>0.77</td><td>103.68</td><td>0.69</td><td>0.59</td><td>78.59</td></tr><tr><th>CHD8</th><td>0</td><td>112</td><td>15</td><td>840</td><td>0.00</td><td>0.24</td><td>32.22</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.16</td><td>20.84</td></tr><tr><th>EPHA3</th><td>5</td><td>107</td><td>20</td><td>835</td><td>1.95</td><td>0.20</td><td>26.47</td><td>1.95</td><td>0.15</td><td>20.55</td><td>1.95</td><td>0.94</td><td>126.08</td></tr><tr><th>HECW2</th><td>2</td><td>110</td><td>13</td><td>842</td><td>1.18</td><td>0.69</td><td>92.36</td><td>1.18</td><td>0.53</td><td>71.52</td><td>1.18</td><td>0.75</td><td>100.90</td></tr><tr><th>IFT140</th><td>2</td><td>110</td><td>7</td><td>848</td><td>2.20</td><td>0.28</td><td>37.55</td><td>2.20</td><td>0.28</td><td>37.55</td><td>2.20</td><td>0.92</td><td>123.89</td></tr><tr><th>IWS1</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>MAGI2</th><td>0</td><td>112</td><td>9</td><td>846</td><td>0.00</td><td>0.61</td><td>81.59</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.33</td><td>44.04</td></tr><tr><th>MAST3</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>MDM4</th><td>5</td><td>107</td><td>21</td><td>834</td><td>1.86</td><td>0.21</td><td>28.44</td><td>1.86</td><td>0.17</td><td>23.22</td><td>1.86</td><td>0.93</td><td>124.65</td></tr><tr><th>MLL5</th><td>3</td><td>109</td><td>9</td><td>846</td><td>2.59</td><td>0.15</td><td>20.53</td><td>2.59</td><td>0.15</td><td>20.53</td><td>2.59</td><td>0.96</td><td>128.58</td></tr><tr><th>PLEKHA8</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>PRDM2</th><td>0</td><td>112</td><td>8</td><td>847</td><td>0.00</td><td>0.61</td><td>81.32</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.37</td><td>49.86</td></tr><tr><th>PRKRIR</th><td>13</td><td>99</td><td>79</td><td>776</td><td>1.29</td><td>0.39</td><td>52.91</td><td>1.29</td><td>0.26</td><td>34.42</td><td>1.29</td><td>0.84</td><td>112.01</td></tr><tr><th>SCAPER</th><td>1</td><td>111</td><td>9</td><td>846</td><td>0.85</td><td>1.00</td><td>134.00</td><td>0.85</td><td>0.71</td><td>95.11</td><td>0.85</td><td>0.67</td><td>90.38</td></tr><tr><th>SETD2</th><td>2</td><td>110</td><td>15</td><td>840</td><td>1.02</td><td>1.00</td><td>134.00</td><td>1.02</td><td>0.60</td><td>80.96</td><td>1.02</td><td>0.69</td><td>91.97</td></tr><tr><th>SMG1</th><td>2</td><td>110</td><td>17</td><td>838</td><td>0.90</td><td>1.00</td><td>134.00</td><td>0.90</td><td>0.67</td><td>89.28</td><td>0.90</td><td>0.62</td><td>82.97</td></tr><tr><th>SMG5</th><td>1</td><td>111</td><td>7</td><td>848</td><td>1.09</td><td>1.00</td><td>134.00</td><td>1.09</td><td>0.63</td><td>84.14</td><td>1.09</td><td>0.77</td><td>102.55</td></tr><tr><th>SMG7</th><td>0</td><td>112</td><td>7</td><td>848</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.42</td><td>56.45</td></tr><tr><th>SNRK</th><td>2</td><td>110</td><td>5</td><td>850</td><td>3.09</td><td>0.19</td><td>25.43</td><td>3.09</td><td>0.19</td><td>25.43</td><td>3.09</td><td>0.96</td><td>128.99</td></tr><tr><th>SPTBN2</th><td>3</td><td>109</td><td>40</td><td>815</td><td>0.56</td><td>0.47</td><td>62.42</td><td>0.56</td><td>0.89</td><td>119.82</td><td>0.56</td><td>0.24</td><td>32.75</td></tr><tr><th>STK11IP</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>STRADA</th><td>1</td><td>111</td><td>6</td><td>849</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.81</td><td>108.57</td></tr><tr><th>SWAP70</th><td>0</td><td>112</td><td>3</td><td>852</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.69</td><td>92.59</td></tr><tr><th>TEP1</th><td>3</td><td>109</td><td>14</td><td>841</td><td>1.65</td><td>0.43</td><td>58.26</td><td>1.65</td><td>0.31</td><td>42.03</td><td>1.65</td><td>0.88</td><td>117.42</td></tr><tr><th>TRIP12</th><td>1</td><td>111</td><td>7</td><td>848</td><td>1.09</td><td>1.00</td><td>134.00</td><td>1.09</td><td>0.63</td><td>84.14</td><td>1.09</td><td>0.77</td><td>102.55</td></tr><tr><th>TTLL5</th><td>1</td><td>111</td><td>9</td><td>846</td><td>0.85</td><td>1.00</td><td>134.00</td><td>0.85</td><td>0.71</td><td>95.11</td><td>0.85</td><td>0.67</td><td>90.38</td></tr><tr><th>UACA</th><td>2</td><td>110</td><td>5</td><td>850</td><td>3.09</td><td>0.19</td><td>25.43</td><td>3.09</td><td>0.19</td><td>25.43</td><td>3.09</td><td>0.96</td><td>128.99</td></tr><tr><th>ZFP91</th><td>0</td><td>112</td><td>10</td><td>845</td><td>0.00</td><td>0.62</td><td>82.51</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.29</td><td>38.89</td></tr><tr><th>ZMIZ1</th><td>7</td><td>105</td><td>19</td><td>836</td><td>2.93</td><td>0.02338</td><td>3.13247</td><td>2.93</td><td>0.02338</td><td>3.13247</td><td>2.93</td><td>0.99</td><td>133.12</td></tr><tr><th>ZNF227</th><td>1</td><td>111</td><td>1</td><td>854</td><td>7.69</td><td>0.22</td><td>29.26</td><td>7.69</td><td>0.22</td><td>29.26</td><td>7.69</td><td>0.99</td><td>132.22</td></tr><tr><th>ZNF668</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>PDS5B</th><td>0</td><td>112</td><td>13</td><td>842</td><td>0.00</td><td>0.38</td><td>51.31</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.20</td><td>26.76</td></tr><tr><th>STAG1</th><td>0</td><td>112</td><td>12</td><td>843</td><td>0.00</td><td>0.38</td><td>50.85</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.23</td><td>30.32</td></tr><tr><th>STAG2</th><td>0</td><td>112</td><td>13</td><td>842</td><td>0.00</td><td>0.38</td><td>51.31</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.20</td><td>26.76</td></tr><tr><th>WAPAL</th><td>0</td><td>112</td><td>2</td><td>853</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.78</td><td>104.74</td></tr><tr><th>PDS5A</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>RAD21</th><td>1</td><td>111</td><td>5</td><td>850</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.85</td><td>114.41</td></tr><tr><th>SMC1A</th><td>0</td><td>112</td><td>11</td><td>844</td><td>0.00</td><td>0.63</td><td>83.94</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.26</td><td>34.34</td></tr><tr><th>KDM6A</th><td>3</td><td>109</td><td>13</td><td>842</td><td>1.78</td><td>0.42</td><td>55.93</td><td>1.78</td><td>0.28</td><td>37.54</td><td>1.78</td><td>0.90</td><td>120.14</td></tr><tr><th>MLL2</th><td>1</td><td>111</td><td>22</td><td>833</td><td>0.34</td><td>0.50</td><td>67.62</td><td>0.34</td><td>0.94</td><td>126.37</td><td>0.34</td><td>0.23</td><td>31.22</td></tr><tr><th>MLL3</th><td>8</td><td>104</td><td>59</td><td>796</td><td>1.04</td><td>0.84</td><td>113.19</td><td>1.04</td><td>0.52</td><td>70.13</td><td>1.04</td><td>0.63</td><td>84.46</td></tr><tr><th>E2F3</th><td>4</td><td>108</td><td>18</td><td>837</td><td>1.72</td><td>0.31</td><td>41.45</td><td>1.72</td><td>0.24</td><td>32.80</td><td>1.72</td><td>0.90</td><td>120.57</td></tr><tr><th>N4BP2</th><td>0</td><td>112</td><td>12</td><td>843</td><td>0.00</td><td>0.38</td><td>50.85</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.23</td><td>30.32</td></tr><tr><th>PROSER1</th><td>0</td><td>112</td><td>7</td><td>848</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.42</td><td>56.45</td></tr><tr><th>ASXL1</th><td>1</td><td>111</td><td>5</td><td>850</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.85</td><td>114.41</td></tr><tr><th>ASXL2</th><td>2</td><td>110</td><td>14</td><td>841</td><td>1.09</td><td>0.71</td><td>94.77</td><td>1.09</td><td>0.57</td><td>76.38</td><td>1.09</td><td>0.72</td><td>96.46</td></tr><tr><th>BAP1</th><td>0</td><td>112</td><td>7</td><td>848</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.42</td><td>56.45</td></tr><tr><th>ANKRD17</th><td>2</td><td>110</td><td>22</td><td>833</td><td>0.69</td><td>1.00</td><td>134.00</td><td>0.69</td><td>0.79</td><td>105.56</td><td>0.69</td><td>0.46</td><td>61.72</td></tr><tr><th>FOXK1</th><td>1</td><td>111</td><td>0</td><td>855</td><td>inf</td><td>0.12</td><td>15.52</td><td>inf</td><td>0.12</td><td>15.52</td><td>inf</td><td>1.00</td><td>134.00</td></tr><tr><th>FOXK2</th><td>1</td><td>111</td><td>3</td><td>852</td><td>2.56</td><td>0.39</td><td>52.17</td><td>2.56</td><td>0.39</td><td>52.17</td><td>2.56</td><td>0.93</td><td>124.86</td></tr><tr><th>KDM1B</th><td>1</td><td>111</td><td>8</td><td>847</td><td>0.95</td><td>1.00</td><td>134.00</td><td>0.95</td><td>0.67</td><td>89.96</td><td>0.95</td><td>0.72</td><td>96.45</td></tr><tr><th>BRAF</th><td>2</td><td>110</td><td>11</td><td>844</td><td>1.40</td><td>0.66</td><td>87.78</td><td>1.40</td><td>0.46</td><td>61.02</td><td>1.40</td><td>0.82</td><td>109.45</td></tr><tr><th>KRAS</th><td>7</td><td>105</td><td>26</td><td>829</td><td>2.13</td><td>0.09</td><td>12.41</td><td>2.13</td><td>0.08</td><td>10.26</td><td>2.13</td><td>0.97</td><td>130.11</td></tr><tr><th>PIK3CA</th><td>29</td><td>83</td><td>302</td><td>553</td><td>0.64</td><td>0.06</td><td>7.54</td><td>0.64</td><td>0.98</td><td>131.71</td><td>0.64</td><td>0.02888</td><td>3.87038</td></tr><tr><th>ARHGAP4</th><td>1</td><td>111</td><td>5</td><td>850</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.52</td><td>70.11</td><td>1.53</td><td>0.85</td><td>114.41</td></tr><tr><th>EIF2AK1</th><td>2</td><td>110</td><td>5</td><td>850</td><td>3.09</td><td>0.19</td><td>25.43</td><td>3.09</td><td>0.19</td><td>25.43</td><td>3.09</td><td>0.96</td><td>128.99</td></tr><tr><th>FAM118B</th><td>2</td><td>110</td><td>6</td><td>849</td><td>2.57</td><td>0.23</td><td>31.45</td><td>2.57</td><td>0.23</td><td>31.45</td><td>2.57</td><td>0.95</td><td>126.65</td></tr><tr><th>NRAS</th><td>0</td><td>112</td><td>3</td><td>852</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.69</td><td>92.59</td></tr><tr><th>PLCE1</th><td>1</td><td>111</td><td>15</td><td>840</td><td>0.50</td><td>1.00</td><td>134.00</td><td>0.50</td><td>0.86</td><td>115.61</td><td>0.50</td><td>0.43</td><td>57.62</td></tr><tr><th>RASGRF2</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>RASGRP2</th><td>1</td><td>111</td><td>6</td><td>849</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.81</td><td>108.57</td></tr><tr><th>RGL1</th><td>0</td><td>112</td><td>2</td><td>853</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.78</td><td>104.74</td></tr><tr><th>VARS2</th><td>0</td><td>112</td><td>1</td><td>854</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.88</td><td>118.48</td></tr><tr><th>B2M</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>CD1D</th><td>0</td><td>112</td><td>6</td><td>849</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.48</td><td>63.89</td></tr><tr><th>HLA-A</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>HLA-B</th><td>1</td><td>111</td><td>7</td><td>848</td><td>1.09</td><td>1.00</td><td>134.00</td><td>1.09</td><td>0.63</td><td>84.14</td><td>1.09</td><td>0.77</td><td>102.55</td></tr><tr><th>P4HTM</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>EGFR</th><td>7</td><td>105</td><td>17</td><td>838</td><td>3.29</td><td>0.01498</td><td>2.00720</td><td>3.29</td><td>0.01498</td><td>2.00720</td><td>3.29</td><td>1.00</td><td>133.49</td></tr><tr><th>ERBB2</th><td>19</td><td>93</td><td>116</td><td>739</td><td>1.30</td><td>0.31</td><td>41.96</td><td>1.30</td><td>0.20</td><td>26.90</td><td>1.30</td><td>0.87</td><td>116.27</td></tr><tr><th>AREG</th><td>3</td><td>109</td><td>9</td><td>846</td><td>2.59</td><td>0.15</td><td>20.53</td><td>2.59</td><td>0.15</td><td>20.53</td><td>2.59</td><td>0.96</td><td>128.58</td></tr><tr><th>ELF3</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>ERBB4</th><td>1</td><td>111</td><td>22</td><td>833</td><td>0.34</td><td>0.50</td><td>67.62</td><td>0.34</td><td>0.94</td><td>126.37</td><td>0.34</td><td>0.23</td><td>31.22</td></tr><tr><th>LRIG1</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>OSMR</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>BOC</th><td>1</td><td>111</td><td>9</td><td>846</td><td>0.85</td><td>1.00</td><td>134.00</td><td>0.85</td><td>0.71</td><td>95.11</td><td>0.85</td><td>0.67</td><td>90.38</td></tr><tr><th>CDON</th><td>2</td><td>110</td><td>7</td><td>848</td><td>2.20</td><td>0.28</td><td>37.55</td><td>2.20</td><td>0.28</td><td>37.55</td><td>2.20</td><td>0.92</td><td>123.89</td></tr><tr><th>KEAP1</th><td>0</td><td>112</td><td>3</td><td>852</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.69</td><td>92.59</td></tr><tr><th>NFE2L2</th><td>2</td><td>110</td><td>4</td><td>851</td><td>3.87</td><td>0.15</td><td>19.59</td><td>3.87</td><td>0.15</td><td>19.59</td><td>3.87</td><td>0.98</td><td>130.88</td></tr><tr><th>CHD6</th><td>3</td><td>109</td><td>17</td><td>838</td><td>1.36</td><td>0.50</td><td>66.60</td><td>1.36</td><td>0.41</td><td>55.47</td><td>1.36</td><td>0.81</td><td>108.13</td></tr><tr><th>PTMA</th><td>0</td><td>112</td><td>1</td><td>854</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.88</td><td>118.48</td></tr><tr><th>WAC</th><td>0</td><td>112</td><td>9</td><td>846</td><td>0.00</td><td>0.61</td><td>81.59</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.33</td><td>44.04</td></tr><tr><th>MYD88</th><td>0</td><td>112</td><td>1</td><td>854</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.88</td><td>118.48</td></tr><tr><th>SPOP</th><td>0</td><td>112</td><td>4</td><td>851</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.61</td><td>81.83</td></tr><tr><th>CBFB</th><td>1</td><td>111</td><td>22</td><td>833</td><td>0.34</td><td>0.50</td><td>67.62</td><td>0.34</td><td>0.94</td><td>126.37</td><td>0.34</td><td>0.23</td><td>31.22</td></tr><tr><th>RUNX1</th><td>1</td><td>111</td><td>31</td><td>824</td><td>0.24</td><td>0.16</td><td>21.94</td><td>0.24</td><td>0.98</td><td>131.56</td><td>0.24</td><td>0.10</td><td>13.03</td></tr><tr><th>ELF4</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>FBXW7</th><td>2</td><td>110</td><td>15</td><td>840</td><td>1.02</td><td>1.00</td><td>134.00</td><td>1.02</td><td>0.60</td><td>80.96</td><td>1.02</td><td>0.69</td><td>91.97</td></tr><tr><th>NOTCH1</th><td>3</td><td>109</td><td>9</td><td>846</td><td>2.59</td><td>0.15</td><td>20.53</td><td>2.59</td><td>0.15</td><td>20.53</td><td>2.59</td><td>0.96</td><td>128.58</td></tr><tr><th>CCNE1</th><td>7</td><td>105</td><td>38</td><td>817</td><td>1.43</td><td>0.35</td><td>46.47</td><td>1.43</td><td>0.26</td><td>34.59</td><td>1.43</td><td>0.86</td><td>115.44</td></tr><tr><th>DCLRE1C</th><td>2</td><td>110</td><td>6</td><td>849</td><td>2.57</td><td>0.23</td><td>31.45</td><td>2.57</td><td>0.23</td><td>31.45</td><td>2.57</td><td>0.95</td><td>126.65</td></tr><tr><th>DLL1</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>DTX4</th><td>1</td><td>111</td><td>7</td><td>848</td><td>1.09</td><td>1.00</td><td>134.00</td><td>1.09</td><td>0.63</td><td>84.14</td><td>1.09</td><td>0.77</td><td>102.55</td></tr><tr><th>FAM101A</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>FOXM1</th><td>1</td><td>111</td><td>6</td><td>849</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.81</td><td>108.57</td></tr><tr><th>JAG1</th><td>2</td><td>110</td><td>11</td><td>844</td><td>1.40</td><td>0.66</td><td>87.78</td><td>1.40</td><td>0.46</td><td>61.02</td><td>1.40</td><td>0.82</td><td>109.45</td></tr><tr><th>KLF5</th><td>0</td><td>112</td><td>0</td><td>855</td><td>nan</td><td>1.00</td><td>134.00</td><td>nan</td><td>1.00</td><td>134.00</td><td>nan</td><td>1.00</td><td>134.00</td></tr><tr><th>LFNG</th><td>0</td><td>112</td><td>1</td><td>854</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.88</td><td>118.48</td></tr><tr><th>LPP</th><td>0</td><td>112</td><td>6</td><td>849</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.48</td><td>63.89</td></tr><tr><th>MAML1</th><td>0</td><td>112</td><td>3</td><td>852</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.69</td><td>92.59</td></tr><tr><th>MMS22L</th><td>0</td><td>112</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.54</td><td>72.31</td></tr><tr><th>NOTCH3</th><td>3</td><td>109</td><td>11</td><td>844</td><td>2.11</td><td>0.21</td><td>28.75</td><td>2.11</td><td>0.21</td><td>28.75</td><td>2.11</td><td>0.93</td><td>124.87</td></tr><tr><th>NOTCH4</th><td>0</td><td>112</td><td>10</td><td>845</td><td>0.00</td><td>0.62</td><td>82.51</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.29</td><td>38.89</td></tr><tr><th>SHPRH</th><td>0</td><td>112</td><td>11</td><td>844</td><td>0.00</td><td>0.63</td><td>83.94</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.26</td><td>34.34</td></tr><tr><th>ARID1A</th><td>2</td><td>110</td><td>25</td><td>830</td><td>0.60</td><td>0.76</td><td>101.83</td><td>0.60</td><td>0.84</td><td>112.63</td><td>0.60</td><td>0.38</td><td>50.54</td></tr><tr><th>PBRM1</th><td>1</td><td>111</td><td>6</td><td>849</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.58</td><td>77.55</td><td>1.27</td><td>0.81</td><td>108.57</td></tr><tr><th>ADNP</th><td>1</td><td>111</td><td>14</td><td>841</td><td>0.54</td><td>1.00</td><td>134.00</td><td>0.54</td><td>0.84</td><td>113.16</td><td>0.54</td><td>0.47</td><td>62.48</td></tr><tr><th>ARID1B</th><td>3</td><td>109</td><td>13</td><td>842</td><td>1.78</td><td>0.42</td><td>55.93</td><td>1.78</td><td>0.28</td><td>37.54</td><td>1.78</td><td>0.90</td><td>120.14</td></tr><tr><th>ARID2</th><td>1</td><td>111</td><td>10</td><td>845</td><td>0.76</td><td>1.00</td><td>134.00</td><td>0.76</td><td>0.74</td><td>99.66</td><td>0.76</td><td>0.63</td><td>84.40</td></tr><tr><th>SMARCB1</th><td>1</td><td>111</td><td>2</td><td>853</td><td>3.84</td><td>0.31</td><td>41.41</td><td>3.84</td><td>0.31</td><td>41.41</td><td>3.84</td><td>0.96</td><td>129.06</td></tr><tr><th>SMARCA4</th><td>0</td><td>112</td><td>7</td><td>848</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.42</td><td>56.45</td></tr><tr><th>NCAPD3</th><td>2</td><td>110</td><td>9</td><td>846</td><td>1.71</td><td>0.37</td><td>49.60</td><td>1.71</td><td>0.37</td><td>49.60</td><td>1.71</td><td>0.87</td><td>117.25</td></tr><tr><th>SMC4</th><td>0</td><td>112</td><td>8</td><td>847</td><td>0.00</td><td>0.61</td><td>81.32</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.37</td><td>49.86</td></tr><tr><th>NCAPG2</th><td>0</td><td>112</td><td>3</td><td>852</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.69</td><td>92.59</td></tr><tr><th>NCAPH2</th><td>0</td><td>112</td><td>3</td><td>852</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>1.00</td><td>134.00</td><td>0.00</td><td>0.69</td><td>92.59</td></tr><tr><th>SMC2</th><td>1</td><td>111</td><td>4</td><td>851</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.46</td><td>61.69</td><td>1.92</td><td>0.89</td><td>119.90</td></tr><tr><th>NCAPD2</th><td>2</td><td>110</td><td>3</td><td>852</td><td>5.16</td><td>0.11</td><td>14.10</td><td>5.16</td><td>0.11</td><td>14.10</td><td>5.16</td><td>0.99</td><td>132.30</td></tr><tr><th>CLASP1</th><td>1</td><td>111</td><td>8</td><td>847</td><td>0.95</td><td>1.00</td><td>134.00</td><td>0.95</td><td>0.67</td><td>89.96</td><td>0.95</td><td>0.72</td><td>96.45</td></tr><tr><th>CLASP2</th><td>2</td><td>110</td><td>6</td><td>849</td><td>2.57</td><td>0.23</td><td>31.45</td><td>2.57</td><td>0.23</td><td>31.45</td><td>2.57</td><td>0.95</td><td>126.65</td></tr><tr><th>CLIP2</th><td>1</td><td>111</td><td>8</td><td>847</td><td>0.95</td><td>1.00</td><td>134.00</td><td>0.95</td><td>0.67</td><td>89.96</td><td>0.95</td><td>0.72</td><td>96.45</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "<IPython.core.display.HTML at 0xc46fc390>"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which pathways are affected in MYC positive samples (Vogelstein)"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "This time use pathway definitions of Vogelstein et al.\n",
      "<p>Note that MYC has been deleted from pathway: Cell Cycle / Apoptosis</p>"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chromatin_modification_pathway=[\"ARID1A\",\"ARID1B\",\"ARID2\",\"ASXL1\",\"ATRX\",\"CREBBP\",\"DAXX\",\"DNMT1\",\"DNMT3A\",\"EP300\",\"EZH2\",\"H3H3A\",\"HIST1H3B\",\"IDH1\",\"IDH2\",\"KDM5C\",\"KDM6A\",\"MEN1\",\"MLL2\",\"MLL3\",\"NCOR1\",\"PAX5\",\"PBRM1\",\"PRDM1\",\"SETD2\",\"SETBP1\",\"SMARCA4\",\"SMARCB1\",\"SPOP\",\"TET2\",\"WT1\",\"NCOA3\"]\n",
      "cell_cycle_apoptosis_pathway=[\"ABL1\",\"BCL2\",\"CARD1\",\"CASP8\",\"CDC73\",\"CDKN2A\",\"CYLD\",\"DAXX\",\"FUBP1\",\"MED12\",\"MYD88\",\"NFE2L2\",\"NPM2\",\"PPP2R1A\",\"RB1\",\"TNFAIP3\",\"TRAF7\",\"TP53\",\"CCND1\",\"CDKN2C\",\"MDM2\",\"MDM4\",\"MYCL1\",\"MYCN\",\"SKP2\"]\n",
      "tgf_beta_pathway=[\"ACVR1B\",\"EP300\",\"FOXL2\",\"GATA1\",\"GATA2\",\"GNAS\",\"MED12\",\"SMAD2\",\"SMAD4\"]\n",
      "PI3K_pathway=[\"AKT1\",\"ALK\",\"B2M\",\"CBL\",\"CEBPA\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"KIT\",\"MET\",\"PDGFRA\",\"PIK2CA\",\"PIK3R1\",\"PTEN\",\"RET\",\"TSC1\",\"TSHR\",\"VHL\",\"NCOA3\"]\n",
      "RAS_pathway=[\"ALK\",\"B2M\",\"BRAF\",\"CBL\",\"CEBPA\",\"CIC\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"HRAS\",\"KIT\",\"KRAS\",\"MAP2K1\",\"MAP3K1\",\"MET\",\"NF1\",\"NRAS\",\"PDGFRA\",\"PTPN11\",\"RET\",\"VHL\"]\n",
      "APC_pathway=[\"AXIN1\",\"CDH1\",\"CTNNB1\",\"EP300\",\"FAM123B\",\"GNAS\",\"HNF1A\",\"NF2\",\"RNF43\",\"SOX9\"]\n",
      "transcription_regulation_pathway=[\"AR\",\"BCOR\",\"CREBBP\",\"GATA3\",\"KLF4\",\"PHF6\",\"RUNX1\",\"SF3B1\",\"SRSF2\",\"U2AF1\",\"IKZF1\",\"LMO1\"]\n",
      "dna_damage_pathway=[\"ATM\",\"BAP1\",\"BRCA1\",\"BRCA2\",\"MLH1\",\"MSH2\",\"MSH6\",\"STAG2\",\"TP53\"]\n",
      "mapkinase_pathway=[\"B2M\",\"CEBPA\",\"GNA11\",\"GNAQ\",\"MAP3K1\",\"TNFAIP3\",\"TSHR\",\"MAP2K4\",\"NKX2-1\"]\n",
      "STAT_pathway=[\"CLRF\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"JAK1\",\"JAK2\",\"JAK3\",\"STAT\",\"MPL\",\"SOCS1\",\"VHL\"]\n",
      "NOTCH_pathway=[\"EP300\",\"FBXW7\",\"GATA1\",\"GATA2\",\"NOTCH1\",\"NOTCH2\"]\n",
      "wnt_pathway=[\"KLF4\"]\n",
      "hedgehog_pathway=[\"PTCH1\",\"SMO\",\"SPOP\"]\n",
      "mTOR_pathway=[\"STK11\"]\n",
      "replicationpathway=[\"SETBP1\"]\n",
      "\n",
      "pathways={\"Chromatin Modification\": chromatin_modification_pathway, \"Cell Cycle\\\\Apoptosis\":cell_cycle_apoptosis_pathway, \n",
      "        \"TGF beta\":tgf_beta_pathway,\"PI3K\":PI3K_pathway,\"RAS\":RAS_pathway, \"APC\":APC_pathway,  \n",
      "        \"Transcription Regulation\":transcription_regulation_pathway, \n",
      "        \"DNA Damage Repair\":dna_damage_pathway, \"MAP Kinase\":mapkinase_pathway,\n",
      "        \"STAT\":STAT_pathway, \"NOTCH\":NOTCH_pathway, \n",
      "        \"WNT\":wnt_pathway, \"Hedgehog\":hedgehog_pathway, \n",
      "        \"mTOR\":mTOR_pathway, \"Replication\":replicationpathway}\n",
      "\n",
      "MYC_pos_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_pos_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,  \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0, \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.0005: \n",
      "      s += \"<td>\"+\"{:.10f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.10f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   elif pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAT</th><td>13</td><td>48</td><td>133</td><td>774</td><td>1.58</td><td>0.19</td><td>2.89</td><td>1.58</td><td>0.11</td><td>1.71</td><td>1.58</td><td>0.94</td><td>14.08</td></tr><tr><th>Chromatin Modification</th><td>28</td><td>33</td><td>268</td><td>639</td><td>2.02</td><td>0.00952</td><td>0.14282</td><td>2.02</td><td>0.00668</td><td>0.10020</td><td>2.02</td><td>1.00</td><td>14.95</td></tr><tr><th>PI3K</th><td>34</td><td>27</td><td>308</td><td>599</td><td>2.45</td><td>0.00082</td><td>0.01230</td><td>2.45</td><td>0.00061</td><td>0.00919</td><td>2.45</td><td>1.00</td><td>15.00</td></tr><tr><th>Transcription Regulation</th><td>12</td><td>49</td><td>175</td><td>732</td><td>1.02</td><td>1.00</td><td>15.00</td><td>1.02</td><td>0.53</td><td>7.89</td><td>1.02</td><td>0.61</td><td>9.08</td></tr><tr><th>NOTCH</th><td>5</td><td>56</td><td>58</td><td>849</td><td>1.31</td><td>0.59</td><td>8.82</td><td>1.31</td><td>0.36</td><td>5.47</td><td>1.31</td><td>0.80</td><td>12.02</td></tr><tr><th>DNA Damage Repair</th><td>42</td><td>19</td><td>311</td><td>596</td><td>4.24</td><td>0.00000</td><td>0.00000</td><td>4.24</td><td>0.0000001131</td><td>0.0000016969</td><td>4.24</td><td>1.00</td><td>15.00</td></tr><tr><th>TGF beta</th><td>3</td><td>58</td><td>70</td><td>837</td><td>0.62</td><td>0.62</td><td>9.23</td><td>0.62</td><td>0.86</td><td>12.85</td><td>0.62</td><td>0.31</td><td>4.61</td></tr><tr><th>MAP Kinase</th><td>10</td><td>51</td><td>141</td><td>766</td><td>1.07</td><td>0.86</td><td>12.83</td><td>1.07</td><td>0.49</td><td>7.31</td><td>1.07</td><td>0.65</td><td>9.78</td></tr><tr><th>Cell Cycle\\Apoptosis</th><td>47</td><td>14</td><td>478</td><td>429</td><td>3.01</td><td>0.00018</td><td>0.00263</td><td>3.01</td><td>0.0001258500</td><td>0.0018877493</td><td>3.01</td><td>1.00</td><td>15.00</td></tr><tr><th>Replication</th><td>0</td><td>61</td><td>13</td><td>894</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>0.43</td><td>6.40</td></tr><tr><th>mTOR</th><td>1</td><td>60</td><td>1</td><td>906</td><td>15.10</td><td>0.12</td><td>1.83</td><td>15.10</td><td>0.12</td><td>1.83</td><td>15.10</td><td>1.00</td><td>14.94</td></tr><tr><th>RAS</th><td>29</td><td>32</td><td>362</td><td>545</td><td>1.36</td><td>0.28</td><td>4.21</td><td>1.36</td><td>0.15</td><td>2.24</td><td>1.36</td><td>0.90</td><td>13.56</td></tr><tr><th>WNT</th><td>0</td><td>61</td><td>2</td><td>905</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>0.88</td><td>13.17</td></tr><tr><th>APC</th><td>8</td><td>53</td><td>156</td><td>751</td><td>0.73</td><td>0.48</td><td>7.26</td><td>0.73</td><td>0.84</td><td>12.62</td><td>0.73</td><td>0.27</td><td>3.98</td></tr><tr><th>Hedgehog</th><td>2</td><td>59</td><td>27</td><td>880</td><td>1.10</td><td>0.70</td><td>10.56</td><td>1.10</td><td>0.56</td><td>8.35</td><td>1.10</td><td>0.73</td><td>10.89</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "<IPython.core.display.HTML at 0xc46eccd0>"
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Remove TP53 from Pathways: Cell Cycle/Apoptosis and DNA Damaga Control"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chromatin_modification_pathway=[\"ARID1A\",\"ARID1B\",\"ARID2\",\"ASXL1\",\"ATRX\",\"CREBBP\",\"DAXX\",\"DNMT1\",\"DNMT3A\",\"EP300\",\"EZH2\",\"H3H3A\",\"HIST1H3B\",\"IDH1\",\"IDH2\",\"KDM5C\",\"KDM6A\",\"MEN1\",\"MLL2\",\"MLL3\",\"NCOR1\",\"PAX5\",\"PBRM1\",\"PRDM1\",\"SETD2\",\"SETBP1\",\"SMARCA4\",\"SMARCB1\",\"SPOP\",\"TET2\",\"WT1\",\"NCOA3\"]\n",
      "cell_cycle_apoptosis_pathway=[\"ABL1\",\"BCL2\",\"CARD1\",\"CASP8\",\"CDC73\",\"CDKN2A\",\"CYLD\",\"DAXX\",\"FUBP1\",\"MED12\",\"MYD88\",\"NFE2L2\",\"NPM2\",\"PPP2R1A\",\"RB1\",\"TNFAIP3\",\"TRAF7\",\"CCND1\",\"CDKN2C\",\"MDM2\",\"MDM4\",\"MYCL1\",\"MYCN\",\"SKP2\"]\n",
      "tgf_beta_pathway=[\"ACVR1B\",\"EP300\",\"FOXL2\",\"GATA1\",\"GATA2\",\"GNAS\",\"MED12\",\"SMAD2\",\"SMAD4\"]\n",
      "PI3K_pathway=[\"AKT1\",\"ALK\",\"B2M\",\"CBL\",\"CEBPA\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"KIT\",\"MET\",\"PDGFRA\",\"PIK2CA\",\"PIK3R1\",\"PTEN\",\"RET\",\"TSC1\",\"TSHR\",\"VHL\",\"NCOA3\"]\n",
      "RAS_pathway=[\"ALK\",\"B2M\",\"BRAF\",\"CBL\",\"CEBPA\",\"CIC\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"HRAS\",\"KIT\",\"KRAS\",\"MAP2K1\",\"MAP3K1\",\"MET\",\"NF1\",\"NRAS\",\"PDGFRA\",\"PTPN11\",\"RET\",\"VHL\"]\n",
      "APC_pathway=[\"AXIN1\",\"CDH1\",\"CTNNB1\",\"EP300\",\"FAM123B\",\"GNAS\",\"HNF1A\",\"NF2\",\"RNF43\",\"SOX9\"]\n",
      "transcription_regulation_pathway=[\"AR\",\"BCOR\",\"CREBBP\",\"GATA3\",\"KLF4\",\"PHF6\",\"RUNX1\",\"SF3B1\",\"SRSF2\",\"U2AF1\",\"IKZF1\",\"LMO1\"]\n",
      "dna_damage_pathway=[\"ATM\",\"BAP1\",\"BRCA1\",\"BRCA2\",\"MLH1\",\"MSH2\",\"MSH6\",\"STAG2\"]\n",
      "mapkinase_pathway=[\"B2M\",\"CEBPA\",\"GNA11\",\"GNAQ\",\"MAP3K1\",\"TNFAIP3\",\"TSHR\",\"MAP2K4\",\"NKX2-1\"]\n",
      "STAT_pathway=[\"CLRF\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"JAK1\",\"JAK2\",\"JAK3\",\"STAT\",\"MPL\",\"SOCS1\",\"VHL\"]\n",
      "NOTCH_pathway=[\"EP300\",\"FBXW7\",\"GATA1\",\"GATA2\",\"NOTCH1\",\"NOTCH2\"]\n",
      "wnt_pathway=[\"KLF4\"]\n",
      "hedgehog_pathway=[\"PTCH1\",\"SMO\",\"SPOP\"]\n",
      "mTOR_pathway=[\"STK11\"]\n",
      "replicationpathway=[\"SETBP1\"]\n",
      "\n",
      "pathways={\"Chromatin Modification\": chromatin_modification_pathway, \"Cell Cycle\\\\Apoptosis\":cell_cycle_apoptosis_pathway, \n",
      "        \"TGF beta\":tgf_beta_pathway,\"PI3K\":PI3K_pathway,\"RAS\":RAS_pathway, \"APC\":APC_pathway,  \n",
      "        \"Transcription Regulation\":transcription_regulation_pathway, \n",
      "        \"DNA Damage Repair\":dna_damage_pathway, \"MAP Kinase\":mapkinase_pathway,\n",
      "        \"STAT\":STAT_pathway, \"NOTCH\":NOTCH_pathway, \n",
      "        \"WNT\":wnt_pathway, \"Hedgehog\":hedgehog_pathway, \n",
      "        \"mTOR\":mTOR_pathway, \"Replication\":replicationpathway}\n",
      "\n",
      "MYC_pos_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_pos_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,  \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0, \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and not sample.checkFocalGeneAmp(\"MYC\"):\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAT</th><td>13</td><td>48</td><td>133</td><td>774</td><td>1.58</td><td>0.19</td><td>2.89</td><td>1.58</td><td>0.11</td><td>1.71</td><td>1.58</td><td>0.94</td><td>14.08</td></tr><tr><th>Chromatin Modification</th><td>28</td><td>33</td><td>268</td><td>639</td><td>2.02</td><td>0.00952</td><td>0.14282</td><td>2.02</td><td>0.00668</td><td>0.10020</td><td>2.02</td><td>1.00</td><td>14.95</td></tr><tr><th>PI3K</th><td>34</td><td>27</td><td>308</td><td>599</td><td>2.45</td><td>0.00082</td><td>0.01230</td><td>2.45</td><td>0.00061</td><td>0.00919</td><td>2.45</td><td>1.00</td><td>15.00</td></tr><tr><th>Transcription Regulation</th><td>12</td><td>49</td><td>175</td><td>732</td><td>1.02</td><td>1.00</td><td>15.00</td><td>1.02</td><td>0.53</td><td>7.89</td><td>1.02</td><td>0.61</td><td>9.08</td></tr><tr><th>NOTCH</th><td>5</td><td>56</td><td>58</td><td>849</td><td>1.31</td><td>0.59</td><td>8.82</td><td>1.31</td><td>0.36</td><td>5.47</td><td>1.31</td><td>0.80</td><td>12.02</td></tr><tr><th>DNA Damage Repair</th><td>9</td><td>52</td><td>74</td><td>833</td><td>1.95</td><td>0.09</td><td>1.40</td><td>1.95</td><td>0.07</td><td>1.03</td><td>1.95</td><td>0.97</td><td>14.56</td></tr><tr><th>TGF beta</th><td>3</td><td>58</td><td>70</td><td>837</td><td>0.62</td><td>0.62</td><td>9.23</td><td>0.62</td><td>0.86</td><td>12.85</td><td>0.62</td><td>0.31</td><td>4.61</td></tr><tr><th>MAP Kinase</th><td>10</td><td>51</td><td>141</td><td>766</td><td>1.07</td><td>0.86</td><td>12.83</td><td>1.07</td><td>0.49</td><td>7.31</td><td>1.07</td><td>0.65</td><td>9.78</td></tr><tr><th>Cell Cycle\\Apoptosis</th><td>26</td><td>35</td><td>317</td><td>590</td><td>1.38</td><td>0.27</td><td>4.02</td><td>1.38</td><td>0.14</td><td>2.12</td><td>1.38</td><td>0.91</td><td>13.66</td></tr><tr><th>Replication</th><td>0</td><td>61</td><td>13</td><td>894</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>0.43</td><td>6.40</td></tr><tr><th>mTOR</th><td>1</td><td>60</td><td>1</td><td>906</td><td>15.10</td><td>0.12</td><td>1.83</td><td>15.10</td><td>0.12</td><td>1.83</td><td>15.10</td><td>1.00</td><td>14.94</td></tr><tr><th>RAS</th><td>29</td><td>32</td><td>362</td><td>545</td><td>1.36</td><td>0.28</td><td>4.21</td><td>1.36</td><td>0.15</td><td>2.24</td><td>1.36</td><td>0.90</td><td>13.56</td></tr><tr><th>WNT</th><td>0</td><td>61</td><td>2</td><td>905</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>0.88</td><td>13.17</td></tr><tr><th>APC</th><td>8</td><td>53</td><td>156</td><td>751</td><td>0.73</td><td>0.48</td><td>7.26</td><td>0.73</td><td>0.84</td><td>12.62</td><td>0.73</td><td>0.27</td><td>3.98</td></tr><tr><th>Hedgehog</th><td>2</td><td>59</td><td>27</td><td>880</td><td>1.10</td><td>0.70</td><td>10.56</td><td>1.10</td><td>0.56</td><td>8.35</td><td>1.10</td><td>0.73</td><td>10.89</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 9,
       "text": [
        "<IPython.core.display.HTML at 0x59e93b50>"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which pathways are affected in MYC copynumber > 4 samples (Vogelstein)"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chromatin_modification_pathway=[\"ARID1A\",\"ARID1B\",\"ARID2\",\"ASXL1\",\"ATRX\",\"CREBBP\",\"DAXX\",\"DNMT1\",\"DNMT3A\",\"EP300\",\"EZH2\",\"H3H3A\",\"HIST1H3B\",\"IDH1\",\"IDH2\",\"KDM5C\",\"KDM6A\",\"MEN1\",\"MLL2\",\"MLL3\",\"NCOR1\",\"PAX5\",\"PBRM1\",\"PRDM1\",\"SETD2\",\"SETBP1\",\"SMARCA4\",\"SMARCB1\",\"SPOP\",\"TET2\",\"WT1\",\"NCOA3\"]\n",
      "cell_cycle_apoptosis_pathway=[\"ABL1\",\"BCL2\",\"CARD1\",\"CASP8\",\"CDC73\",\"CDKN2A\",\"CYLD\",\"DAXX\",\"FUBP1\",\"MED12\",\"MYD88\",\"NFE2L2\",\"NPM2\",\"PPP2R1A\",\"RB1\",\"TNFAIP3\",\"TRAF7\",\"TP53\",\"CCND1\",\"CDKN2C\",\"MDM2\",\"MDM4\",\"MYCL1\",\"MYCN\",\"SKP2\"]\n",
      "tgf_beta_pathway=[\"ACVR1B\",\"EP300\",\"FOXL2\",\"GATA1\",\"GATA2\",\"GNAS\",\"MED12\",\"SMAD2\",\"SMAD4\"]\n",
      "PI3K_pathway=[\"AKT1\",\"ALK\",\"B2M\",\"CBL\",\"CEBPA\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"KIT\",\"MET\",\"PDGFRA\",\"PIK2CA\",\"PIK3R1\",\"PTEN\",\"RET\",\"TSC1\",\"TSHR\",\"VHL\",\"NCOA3\"]\n",
      "RAS_pathway=[\"ALK\",\"B2M\",\"BRAF\",\"CBL\",\"CEBPA\",\"CIC\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"HRAS\",\"KIT\",\"KRAS\",\"MAP2K1\",\"MAP3K1\",\"MET\",\"NF1\",\"NRAS\",\"PDGFRA\",\"PTPN11\",\"RET\",\"VHL\"]\n",
      "APC_pathway=[\"AXIN1\",\"CDH1\",\"CTNNB1\",\"EP300\",\"FAM123B\",\"GNAS\",\"HNF1A\",\"NF2\",\"RNF43\",\"SOX9\"]\n",
      "transcription_regulation_pathway=[\"AR\",\"BCOR\",\"CREBBP\",\"GATA3\",\"KLF4\",\"PHF6\",\"RUNX1\",\"SF3B1\",\"SRSF2\",\"U2AF1\",\"IKZF1\",\"LMO1\"]\n",
      "dna_damage_pathway=[\"ATM\",\"BAP1\",\"BRCA1\",\"BRCA2\",\"MLH1\",\"MSH2\",\"MSH6\",\"STAG2\",\"TP53\"]\n",
      "mapkinase_pathway=[\"B2M\",\"CEBPA\",\"GNA11\",\"GNAQ\",\"MAP3K1\",\"TNFAIP3\",\"TSHR\",\"MAP2K4\",\"NKX2-1\"]\n",
      "STAT_pathway=[\"CLRF\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"JAK1\",\"JAK2\",\"JAK3\",\"STAT\",\"MPL\",\"SOCS1\",\"VHL\"]\n",
      "NOTCH_pathway=[\"EP300\",\"FBXW7\",\"GATA1\",\"GATA2\",\"NOTCH1\",\"NOTCH2\"]\n",
      "wnt_pathway=[\"KLF4\"]\n",
      "hedgehog_pathway=[\"PTCH1\",\"SMO\",\"SPOP\"]\n",
      "mTOR_pathway=[\"STK11\"]\n",
      "replicationpathway=[\"SETBP1\"]\n",
      "\n",
      "pathways={\"Chromatin Modification\": chromatin_modification_pathway, \"Cell Cycle\\\\Apoptosis\":cell_cycle_apoptosis_pathway, \n",
      "        \"TGF beta\":tgf_beta_pathway,\"PI3K\":PI3K_pathway,\"RAS\":RAS_pathway, \"APC\":APC_pathway,  \n",
      "        \"Transcription Regulation\":transcription_regulation_pathway, \n",
      "        \"DNA Damage Repair\":dna_damage_pathway, \"MAP Kinase\":mapkinase_pathway,\n",
      "        \"STAT\":STAT_pathway, \"NOTCH\":NOTCH_pathway, \n",
      "        \"WNT\":wnt_pathway, \"Hedgehog\":hedgehog_pathway, \n",
      "        \"mTOR\":mTOR_pathway, \"Replication\":replicationpathway}\n",
      "\n",
      "MYC_pos_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_pos_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,  \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0, \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    copynumber = \"n/a\"\n",
      "    log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "    if (log2 != None):\n",
      "        copynumber = 2.0 * 2.0 ** log2\n",
      "    else:\n",
      "        copynumber=\"n/a\"\n",
      "    if copynumber == \"n/a\":\n",
      "        continue\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if pathway_affected and copynumber > 4:\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and copynumber > 4:\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif pathway_affected and not copynumber > 4:\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and not copynumber > 4:\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.0005: \n",
      "      s += \"<td>\"+\"{:.10f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.10f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   elif pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAT</th><td>21</td><td>91</td><td>125</td><td>730</td><td>1.35</td><td>0.26</td><td>3.92</td><td>1.35</td><td>0.16</td><td>2.35</td><td>1.35</td><td>0.90</td><td>13.48</td></tr><tr><th>Chromatin Modification</th><td>47</td><td>65</td><td>249</td><td>606</td><td>1.76</td><td>0.00857</td><td>0.12859</td><td>1.76</td><td>0.00451</td><td>0.06767</td><td>1.76</td><td>1.00</td><td>14.96</td></tr><tr><th>PI3K</th><td>49</td><td>63</td><td>293</td><td>562</td><td>1.49</td><td>0.06</td><td>0.87</td><td>1.49</td><td>0.03201</td><td>0.48010</td><td>1.49</td><td>0.98</td><td>14.70</td></tr><tr><th>Transcription Regulation</th><td>18</td><td>94</td><td>168</td><td>687</td><td>0.78</td><td>0.44</td><td>6.66</td><td>0.78</td><td>0.85</td><td>12.74</td><td>0.78</td><td>0.22</td><td>3.32</td></tr><tr><th>NOTCH</th><td>10</td><td>102</td><td>53</td><td>802</td><td>1.48</td><td>0.31</td><td>4.58</td><td>1.48</td><td>0.18</td><td>2.73</td><td>1.48</td><td>0.90</td><td>13.50</td></tr><tr><th>DNA Damage Repair</th><td>68</td><td>44</td><td>284</td><td>571</td><td>3.11</td><td>0.00000</td><td>0.00000</td><td>3.11</td><td>0.0000000228</td><td>0.0000003415</td><td>3.11</td><td>1.00</td><td>15.00</td></tr><tr><th>TGF beta</th><td>11</td><td>101</td><td>62</td><td>793</td><td>1.39</td><td>0.34</td><td>5.11</td><td>1.39</td><td>0.21</td><td>3.20</td><td>1.39</td><td>0.87</td><td>13.12</td></tr><tr><th>MAP Kinase</th><td>18</td><td>94</td><td>133</td><td>722</td><td>1.04</td><td>0.89</td><td>13.35</td><td>1.04</td><td>0.49</td><td>7.33</td><td>1.04</td><td>0.62</td><td>9.28</td></tr><tr><th>Cell Cycle\\Apoptosis</th><td>87</td><td>25</td><td>437</td><td>418</td><td>3.33</td><td>0.00000</td><td>0.00000</td><td>3.33</td><td>0.0000000420</td><td>0.0000006296</td><td>3.33</td><td>1.00</td><td>15.00</td></tr><tr><th>Replication</th><td>1</td><td>111</td><td>12</td><td>843</td><td>0.63</td><td>1.00</td><td>15.00</td><td>0.63</td><td>0.80</td><td>12.00</td><td>0.63</td><td>0.54</td><td>8.17</td></tr><tr><th>mTOR</th><td>1</td><td>111</td><td>1</td><td>854</td><td>7.69</td><td>0.22</td><td>3.28</td><td>7.69</td><td>0.22</td><td>3.28</td><td>7.69</td><td>0.99</td><td>14.80</td></tr><tr><th>RAS</th><td>52</td><td>60</td><td>338</td><td>517</td><td>1.33</td><td>0.18</td><td>2.74</td><td>1.33</td><td>0.10</td><td>1.47</td><td>1.33</td><td>0.93</td><td>13.99</td></tr><tr><th>WNT</th><td>0</td><td>112</td><td>2</td><td>853</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>0.78</td><td>11.72</td></tr><tr><th>APC</th><td>15</td><td>97</td><td>149</td><td>706</td><td>0.73</td><td>0.35</td><td>5.23</td><td>0.73</td><td>0.89</td><td>13.32</td><td>0.73</td><td>0.18</td><td>2.63</td></tr><tr><th>Hedgehog</th><td>4</td><td>108</td><td>25</td><td>830</td><td>1.23</td><td>0.77</td><td>11.48</td><td>1.23</td><td>0.44</td><td>6.57</td><td>1.23</td><td>0.76</td><td>11.43</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 10,
       "text": [
        "<IPython.core.display.HTML at 0xc4707790>"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Remove TP53 from pathways Cell Cycle / Apoptosis and DNA Damage repair"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chromatin_modification_pathway=[\"ARID1A\",\"ARID1B\",\"ARID2\",\"ASXL1\",\"ATRX\",\"CREBBP\",\"DAXX\",\"DNMT1\",\"DNMT3A\",\"EP300\",\"EZH2\",\"H3H3A\",\"HIST1H3B\",\"IDH1\",\"IDH2\",\"KDM5C\",\"KDM6A\",\"MEN1\",\"MLL2\",\"MLL3\",\"NCOR1\",\"PAX5\",\"PBRM1\",\"PRDM1\",\"SETD2\",\"SETBP1\",\"SMARCA4\",\"SMARCB1\",\"SPOP\",\"TET2\",\"WT1\",\"NCOA3\"]\n",
      "cell_cycle_apoptosis_pathway=[\"ABL1\",\"BCL2\",\"CARD1\",\"CASP8\",\"CDC73\",\"CDKN2A\",\"CYLD\",\"DAXX\",\"FUBP1\",\"MED12\",\"MYD88\",\"NFE2L2\",\"NPM2\",\"PPP2R1A\",\"RB1\",\"TNFAIP3\",\"TRAF7\",\"CCND1\",\"CDKN2C\",\"MDM2\",\"MDM4\",\"MYCL1\",\"MYCN\",\"SKP2\"]\n",
      "tgf_beta_pathway=[\"ACVR1B\",\"EP300\",\"FOXL2\",\"GATA1\",\"GATA2\",\"GNAS\",\"MED12\",\"SMAD2\",\"SMAD4\"]\n",
      "PI3K_pathway=[\"AKT1\",\"ALK\",\"B2M\",\"CBL\",\"CEBPA\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"KIT\",\"MET\",\"PDGFRA\",\"PIK2CA\",\"PIK3R1\",\"PTEN\",\"RET\",\"TSC1\",\"TSHR\",\"VHL\",\"NCOA3\"]\n",
      "RAS_pathway=[\"ALK\",\"B2M\",\"BRAF\",\"CBL\",\"CEBPA\",\"CIC\",\"CSF1R\",\"EGFR\",\"ERBB2\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"GNA11\",\"GNAQ\",\"GNAS\",\"HRAS\",\"KIT\",\"KRAS\",\"MAP2K1\",\"MAP3K1\",\"MET\",\"NF1\",\"NRAS\",\"PDGFRA\",\"PTPN11\",\"RET\",\"VHL\"]\n",
      "APC_pathway=[\"AXIN1\",\"CDH1\",\"CTNNB1\",\"EP300\",\"FAM123B\",\"GNAS\",\"HNF1A\",\"NF2\",\"RNF43\",\"SOX9\"]\n",
      "transcription_regulation_pathway=[\"AR\",\"BCOR\",\"CREBBP\",\"GATA3\",\"KLF4\",\"PHF6\",\"RUNX1\",\"SF3B1\",\"SRSF2\",\"U2AF1\",\"IKZF1\",\"LMO1\"]\n",
      "dna_damage_pathway=[\"ATM\",\"BAP1\",\"BRCA1\",\"BRCA2\",\"MLH1\",\"MSH2\",\"MSH6\",\"STAG2\"]\n",
      "mapkinase_pathway=[\"B2M\",\"CEBPA\",\"GNA11\",\"GNAQ\",\"MAP3K1\",\"TNFAIP3\",\"TSHR\",\"MAP2K4\",\"NKX2-1\"]\n",
      "STAT_pathway=[\"CLRF\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"JAK1\",\"JAK2\",\"JAK3\",\"STAT\",\"MPL\",\"SOCS1\",\"VHL\"]\n",
      "NOTCH_pathway=[\"EP300\",\"FBXW7\",\"GATA1\",\"GATA2\",\"NOTCH1\",\"NOTCH2\"]\n",
      "wnt_pathway=[\"KLF4\"]\n",
      "hedgehog_pathway=[\"PTCH1\",\"SMO\",\"SPOP\"]\n",
      "mTOR_pathway=[\"STK11\"]\n",
      "replicationpathway=[\"SETBP1\"]\n",
      "\n",
      "pathways={\"Chromatin Modification\": chromatin_modification_pathway, \"Cell Cycle\\\\Apoptosis\":cell_cycle_apoptosis_pathway, \n",
      "        \"TGF beta\":tgf_beta_pathway,\"PI3K\":PI3K_pathway,\"RAS\":RAS_pathway, \"APC\":APC_pathway,  \n",
      "        \"Transcription Regulation\":transcription_regulation_pathway, \n",
      "        \"DNA Damage Repair\":dna_damage_pathway, \"MAP Kinase\":mapkinase_pathway,\n",
      "        \"STAT\":STAT_pathway, \"NOTCH\":NOTCH_pathway, \n",
      "        \"WNT\":wnt_pathway, \"Hedgehog\":hedgehog_pathway, \n",
      "        \"mTOR\":mTOR_pathway, \"Replication\":replicationpathway}\n",
      "\n",
      "MYC_pos_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_pos_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,\n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_affected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0,  \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "MYC_neg_pathway_unaffected={\"Chromatin Modification\": 0, \"Cell Cycle\\\\Apoptosis\":0, \"TGF beta\":0,\"PI3K\":0,\"RAS\":0, \"APC\":0, \n",
      "                          \"Transcription Regulation\":0, \"DNA Damage Repair\":0, \"MAP Kinase\":0,\"STAT\":0, \"NOTCH\":0, \"WNT\":0, \"Hedgehog\":0,\n",
      "                          \"mTOR\":0, \"Replication\":0}\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    copynumber = \"n/a\"\n",
      "    log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "    if (log2 != None):\n",
      "        copynumber = 2.0 * 2.0 ** log2\n",
      "    else:\n",
      "        copynumber=\"n/a\"\n",
      "    if copynumber == \"n/a\":\n",
      "        continue\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if pathway_affected and copynumber > 4:\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and copynumber > 4:\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif pathway_affected and not copynumber > 4:\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif not pathway_affected and not copynumber > 4:\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 15)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 15)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAT</th><td>21</td><td>91</td><td>125</td><td>730</td><td>1.35</td><td>0.26</td><td>3.92</td><td>1.35</td><td>0.16</td><td>2.35</td><td>1.35</td><td>0.90</td><td>13.48</td></tr><tr><th>Chromatin Modification</th><td>47</td><td>65</td><td>249</td><td>606</td><td>1.76</td><td>0.00857</td><td>0.12859</td><td>1.76</td><td>0.00451</td><td>0.06767</td><td>1.76</td><td>1.00</td><td>14.96</td></tr><tr><th>PI3K</th><td>49</td><td>63</td><td>293</td><td>562</td><td>1.49</td><td>0.06</td><td>0.87</td><td>1.49</td><td>0.03201</td><td>0.48010</td><td>1.49</td><td>0.98</td><td>14.70</td></tr><tr><th>Transcription Regulation</th><td>18</td><td>94</td><td>168</td><td>687</td><td>0.78</td><td>0.44</td><td>6.66</td><td>0.78</td><td>0.85</td><td>12.74</td><td>0.78</td><td>0.22</td><td>3.32</td></tr><tr><th>NOTCH</th><td>10</td><td>102</td><td>53</td><td>802</td><td>1.48</td><td>0.31</td><td>4.58</td><td>1.48</td><td>0.18</td><td>2.73</td><td>1.48</td><td>0.90</td><td>13.50</td></tr><tr><th>DNA Damage Repair</th><td>9</td><td>103</td><td>74</td><td>781</td><td>0.92</td><td>1.00</td><td>15.00</td><td>0.92</td><td>0.64</td><td>9.64</td><td>0.92</td><td>0.50</td><td>7.49</td></tr><tr><th>TGF beta</th><td>11</td><td>101</td><td>62</td><td>793</td><td>1.39</td><td>0.34</td><td>5.11</td><td>1.39</td><td>0.21</td><td>3.20</td><td>1.39</td><td>0.87</td><td>13.12</td></tr><tr><th>MAP Kinase</th><td>18</td><td>94</td><td>133</td><td>722</td><td>1.04</td><td>0.89</td><td>13.35</td><td>1.04</td><td>0.49</td><td>7.33</td><td>1.04</td><td>0.62</td><td>9.28</td></tr><tr><th>Cell Cycle\\Apoptosis</th><td>50</td><td>62</td><td>293</td><td>562</td><td>1.55</td><td>0.03556</td><td>0.53343</td><td>1.55</td><td>0.02106</td><td>0.31585</td><td>1.55</td><td>0.99</td><td>14.81</td></tr><tr><th>Replication</th><td>1</td><td>111</td><td>12</td><td>843</td><td>0.63</td><td>1.00</td><td>15.00</td><td>0.63</td><td>0.80</td><td>12.00</td><td>0.63</td><td>0.54</td><td>8.17</td></tr><tr><th>mTOR</th><td>1</td><td>111</td><td>1</td><td>854</td><td>7.69</td><td>0.22</td><td>3.28</td><td>7.69</td><td>0.22</td><td>3.28</td><td>7.69</td><td>0.99</td><td>14.80</td></tr><tr><th>RAS</th><td>52</td><td>60</td><td>338</td><td>517</td><td>1.33</td><td>0.18</td><td>2.74</td><td>1.33</td><td>0.10</td><td>1.47</td><td>1.33</td><td>0.93</td><td>13.99</td></tr><tr><th>WNT</th><td>0</td><td>112</td><td>2</td><td>853</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>1.00</td><td>15.00</td><td>0.00</td><td>0.78</td><td>11.72</td></tr><tr><th>APC</th><td>15</td><td>97</td><td>149</td><td>706</td><td>0.73</td><td>0.35</td><td>5.23</td><td>0.73</td><td>0.89</td><td>13.32</td><td>0.73</td><td>0.18</td><td>2.63</td></tr><tr><th>Hedgehog</th><td>4</td><td>108</td><td>25</td><td>830</td><td>1.23</td><td>0.77</td><td>11.48</td><td>1.23</td><td>0.44</td><td>6.57</td><td>1.23</td><td>0.76</td><td>11.43</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 11,
       "text": [
        "<IPython.core.display.HTML at 0x59ea0310>"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which Genes are affected in MYC positive samples (Vogelstein)"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "This time use genes of pathway definitions of Vogelstein et al.\n",
      "<p>Note that MYC has been deleted from the gene list</p>"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes=[\"ABL1\",\"ACVR1B\",\"AKT1\",\"ALK\",\"APC\",\"AR\",\"ARID1A\",\"ARID1B\",\"ARID2\",\"ASXL1\",\"ATM\",\"ATRX\",\"AXIN1\",\"B2M\",\"BAP1\",\"BCL2\",\"BCOR\",\"BRAF\",\n",
      "       \"BRCA1\",\"BRCA2\",\"CARD11\",\"CASP8\",\"CBL\",\"CDC73\",\"CDH1\",\"CDKN2A\",\"CEBPA\",\"CIC\",\"CREBBP\",\"CRLF2\",\"CSF1R\",\"CTNNB1\",\"CYLD\",\"DAXX\",\n",
      "       \"DNMT1\",\"DNMT3A\",\"EGFR\",\"EP300\",\"ERBB2\",\"EZH2\",\"FAM123B\",\"FBXW7\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"FOXL2\",\"FUBP1\",\"GATA1\",\"GATA2\",\"GATA3\",\n",
      "       \"GNA11\",\"GNAQ\",\"GNAS\",\"H3F3A\",\"HIST1H3B\",\"HNF1A\",\"HRAS\",\"IDH1\",\"IDH2\",\"JAK1\",\"JAK2\",\"JAK3\",\"KDM5C\",\"KDM6A\",\"KIT\",\"KLF4\",\"KRAS\",\n",
      "       \"MAP2K1\",\"MAP3K1\",\"MED12\",\"MEN1\",\"MET\",\"MLH1\",\"MLL2\",\"MLL3\",\"MPL\",\"MSH2\",\"MSH6\",\"MYD88\",\"NCOR1\",\"NF1\",\"NF2\",\"NFE2L2\",\"NOTCH1\",\n",
      "       \"NOTCH2\",\"NPM1\",\"NRAS\",\"PAX5\",\"PBRM1\",\"PDGFRA\",\"PHF6\",\"PIK3CA\",\"PIK3R1\",\"PPP2R1A\",\"PRDM1\",\"PTCH1\",\"PTEN\",\"PTPN11\",\"RB1\",\"RET\",\n",
      "       \"RNF43\",\"RUNX1\",\"SETD2\",\"SETBP1\",\"SF3B1\",\"SMAD2\",\"SMAD4\",\"SMARCA4\",\"SMARCB1\",\"SMO\",\"SOCS1\",\"SOX9\",\"SPOP\",\"SRSF2\",\"STAG2\",\"STK11\",\n",
      "       \"TET2\",\"TNFAIP3\",\"TRAF7\",\"TP53\",\"TSC1\",\"TSHR\",\"U2AF1\",\"VHL\",\"WT1\",\"CCND1\",\"CDKN2C\",\"IKZF1\",\"LMO1\",\"MAP2K4\",\"MDM2\",\"MDM4\",\n",
      "       \"MYCL1\",\"MYCN\",\"NCOA3\",\"NKX2-1\",\"SKP2\"]\n",
      "\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "gene_count = 0\n",
      "for gene in genes:\n",
      "    gene_count += 1\n",
      "    MYC_pos_gene_affected = 0\n",
      "    MYC_pos_gene_unaffected = 0\n",
      "    MYC_neg_gene_affected = 0\n",
      "    MYC_neg_gene_unaffected = 0\n",
      "\n",
      "    for sample in samples.values():\n",
      "        if not sample.somatic_mutation_data:\n",
      "            continue\n",
      "        if sample.checkFocalGeneAmp(gene) or gene in sample.genes_affected:\n",
      "            if sample.checkFocalGeneAmp(\"MYC\"):\n",
      "                MYC_pos_gene_affected += 1\n",
      "            else:\n",
      "                MYC_neg_gene_affected += 1\n",
      "        else:\n",
      "            if sample.checkFocalGeneAmp(\"MYC\"):\n",
      "                MYC_pos_gene_unaffected += 1\n",
      "            else:\n",
      "                MYC_neg_gene_unaffected += 1\n",
      "\n",
      "    oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]])\n",
      "    oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='greater')\n",
      "    oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='less')\n",
      "\n",
      "    if (pvalue > 0.05 and pvalue_greater > 0.05 and pvalue_less > 0.05):\n",
      "        continue\n",
      "        \n",
      "    s += \"<tr><th>\"+gene+\"</th><td>\"+str(MYC_pos_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_pos_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "    if pvalue < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue * 137)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue * 137)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "    if pvalue_greater < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 137)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 137)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "    if pvalue_less < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 137)+\"</td></tr>\"\n",
      "    else: \n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 137)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "print \"Genes: \"+str(gene_count)\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Genes: 137\n"
       ]
      },
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>AXIN1</th><td>2</td><td>59</td><td>3</td><td>904</td><td>10.21</td><td>0.03454</td><td>4.73142</td><td>10.21</td><td>0.03454</td><td>4.73142</td><td>10.21</td><td>1.00</td><td>136.70</td></tr><tr><th>BRCA1</th><td>5</td><td>56</td><td>10</td><td>897</td><td>8.01</td><td>0.00155</td><td>0.21278</td><td>8.01</td><td>0.00155</td><td>0.21278</td><td>8.01</td><td>1.00</td><td>136.98</td></tr><tr><th>CEBPA</th><td>4</td><td>57</td><td>4</td><td>903</td><td>15.84</td><td>0.00083</td><td>0.11344</td><td>15.84</td><td>0.00083</td><td>0.11344</td><td>15.84</td><td>1.00</td><td>136.99</td></tr><tr><th>CYLD</th><td>3</td><td>58</td><td>8</td><td>899</td><td>5.81</td><td>0.02733</td><td>3.74448</td><td>5.81</td><td>0.02733</td><td>3.74448</td><td>5.81</td><td>1.00</td><td>136.54</td></tr><tr><th>ERBB2</th><td>15</td><td>46</td><td>120</td><td>787</td><td>2.14</td><td>0.02042</td><td>2.79781</td><td>2.14</td><td>0.01532</td><td>2.09915</td><td>2.14</td><td>0.99</td><td>136.13</td></tr><tr><th>PIK3CA</th><td>14</td><td>47</td><td>317</td><td>590</td><td>0.55</td><td>0.07</td><td>9.46</td><td>0.55</td><td>0.98</td><td>134.59</td><td>0.55</td><td>0.03528</td><td>4.83290</td></tr><tr><th>SOCS1</th><td>2</td><td>59</td><td>4</td><td>903</td><td>7.65</td><td>0.04972</td><td>6.81214</td><td>7.65</td><td>0.04972</td><td>6.81214</td><td>7.65</td><td>1.00</td><td>136.43</td></tr><tr><th>TP53</th><td>36</td><td>25</td><td>261</td><td>646</td><td>3.56</td><td>0.00000</td><td>0.00033</td><td>3.56</td><td>0.00000</td><td>0.00028</td><td>3.56</td><td>1.00</td><td>137.00</td></tr><tr><th>NCOA3</th><td>7</td><td>54</td><td>16</td><td>891</td><td>7.22</td><td>0.00031</td><td>0.04229</td><td>7.22</td><td>0.00031</td><td>0.04229</td><td>7.22</td><td>1.00</td><td>137.00</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 12,
       "text": [
        "<IPython.core.display.HTML at 0x59efb8d0>"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which Genes are affected in MYC copynumber > 4 samples (Vogelstein)"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes=[\"ABL1\",\"ACVR1B\",\"AKT1\",\"ALK\",\"APC\",\"AR\",\"ARID1A\",\"ARID1B\",\"ARID2\",\"ASXL1\",\"ATM\",\"ATRX\",\"AXIN1\",\"B2M\",\"BAP1\",\"BCL2\",\"BCOR\",\"BRAF\",\n",
      "       \"BRCA1\",\"BRCA2\",\"CARD11\",\"CASP8\",\"CBL\",\"CDC73\",\"CDH1\",\"CDKN2A\",\"CEBPA\",\"CIC\",\"CREBBP\",\"CRLF2\",\"CSF1R\",\"CTNNB1\",\"CYLD\",\"DAXX\",\n",
      "       \"DNMT1\",\"DNMT3A\",\"EGFR\",\"EP300\",\"ERBB2\",\"EZH2\",\"FAM123B\",\"FBXW7\",\"FGFR2\",\"FGFR3\",\"FLT3\",\"FOXL2\",\"FUBP1\",\"GATA1\",\"GATA2\",\"GATA3\",\n",
      "       \"GNA11\",\"GNAQ\",\"GNAS\",\"H3F3A\",\"HIST1H3B\",\"HNF1A\",\"HRAS\",\"IDH1\",\"IDH2\",\"JAK1\",\"JAK2\",\"JAK3\",\"KDM5C\",\"KDM6A\",\"KIT\",\"KLF4\",\"KRAS\",\n",
      "       \"MAP2K1\",\"MAP3K1\",\"MED12\",\"MEN1\",\"MET\",\"MLH1\",\"MLL2\",\"MLL3\",\"MPL\",\"MSH2\",\"MSH6\",\"MYD88\",\"NCOR1\",\"NF1\",\"NF2\",\"NFE2L2\",\"NOTCH1\",\n",
      "       \"NOTCH2\",\"NPM1\",\"NRAS\",\"PAX5\",\"PBRM1\",\"PDGFRA\",\"PHF6\",\"PIK3CA\",\"PIK3R1\",\"PPP2R1A\",\"PRDM1\",\"PTCH1\",\"PTEN\",\"PTPN11\",\"RB1\",\"RET\",\n",
      "       \"RNF43\",\"RUNX1\",\"SETD2\",\"SETBP1\",\"SF3B1\",\"SMAD2\",\"SMAD4\",\"SMARCA4\",\"SMARCB1\",\"SMO\",\"SOCS1\",\"SOX9\",\"SPOP\",\"SRSF2\",\"STAG2\",\"STK11\",\n",
      "       \"TET2\",\"TNFAIP3\",\"TRAF7\",\"TP53\",\"TSC1\",\"TSHR\",\"U2AF1\",\"VHL\",\"WT1\",\"CCND1\",\"CDKN2C\",\"IKZF1\",\"LMO1\",\"MAP2K4\",\"MDM2\",\"MDM4\",\n",
      "       \"MYCL1\",\"MYCN\",\"NCOA3\",\"NKX2-1\",\"SKP2\"]\n",
      "\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "gene_count = 0\n",
      "for gene in genes:\n",
      "    gene_count += 1\n",
      "    MYC_pos_gene_affected = 0\n",
      "    MYC_pos_gene_unaffected = 0\n",
      "    MYC_neg_gene_affected = 0\n",
      "    MYC_neg_gene_unaffected = 0\n",
      "\n",
      "    for sample in samples.values():\n",
      "        if not sample.somatic_mutation_data:\n",
      "            continue\n",
      "        copynumber = \"n/a\"\n",
      "        log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "        if (log2 != None):\n",
      "           copynumber = 2.0 * 2.0 ** log2\n",
      "        else:\n",
      "           copynumber=\"n/a\"\n",
      "        if copynumber == \"n/a\":\n",
      "           continue\n",
      "        if sample.checkFocalGeneAmp(gene) or gene in sample.genes_affected:\n",
      "            if copynumber > 4:\n",
      "                MYC_pos_gene_affected += 1\n",
      "            else:\n",
      "                MYC_neg_gene_affected += 1\n",
      "        else:\n",
      "            if copynumber > 4:\n",
      "                MYC_pos_gene_unaffected += 1\n",
      "            else:\n",
      "                MYC_neg_gene_unaffected += 1\n",
      "\n",
      "    oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]])\n",
      "    oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='greater')\n",
      "    oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='less')\n",
      "\n",
      "    if (pvalue > 0.05 and pvalue_greater > 0.05 and pvalue_less > 0.05):\n",
      "        continue\n",
      "        \n",
      "    s += \"<tr><th>\"+gene+\"</th><td>\"+str(MYC_pos_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_pos_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "    if pvalue < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue * 137)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue * 137)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "    if pvalue_greater < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 137)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 137)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "    if pvalue_less < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 137)+\"</td></tr>\"\n",
      "    else: \n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 137)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "print \"Genes: \"+str(gene_count)\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Genes: 137\n"
       ]
      },
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>AXIN1</th><td>3</td><td>109</td><td>2</td><td>853</td><td>11.74</td><td>0.01271</td><td>1.74168</td><td>11.74</td><td>0.01271</td><td>1.74168</td><td>11.74</td><td>1.00</td><td>136.89</td></tr><tr><th>CDH1</th><td>7</td><td>105</td><td>105</td><td>750</td><td>0.48</td><td>0.06</td><td>8.29</td><td>0.48</td><td>0.98</td><td>134.91</td><td>0.48</td><td>0.03617</td><td>4.95570</td></tr><tr><th>CYLD</th><td>6</td><td>106</td><td>5</td><td>850</td><td>9.62</td><td>0.00060</td><td>0.08254</td><td>9.62</td><td>0.00060</td><td>0.08254</td><td>9.62</td><td>1.00</td><td>136.99</td></tr><tr><th>EGFR</th><td>7</td><td>105</td><td>17</td><td>838</td><td>3.29</td><td>0.01498</td><td>2.05213</td><td>3.29</td><td>0.01498</td><td>2.05213</td><td>3.29</td><td>1.00</td><td>136.48</td></tr><tr><th>FGFR3</th><td>6</td><td>106</td><td>15</td><td>840</td><td>3.17</td><td>0.02661</td><td>3.64589</td><td>3.17</td><td>0.02661</td><td>3.64589</td><td>3.17</td><td>0.99</td><td>136.07</td></tr><tr><th>PIK3CA</th><td>29</td><td>83</td><td>302</td><td>553</td><td>0.64</td><td>0.06</td><td>7.71</td><td>0.64</td><td>0.98</td><td>134.66</td><td>0.64</td><td>0.02888</td><td>3.95703</td></tr><tr><th>SOCS1</th><td>3</td><td>109</td><td>3</td><td>852</td><td>7.82</td><td>0.02329</td><td>3.19118</td><td>7.82</td><td>0.02329</td><td>3.19118</td><td>7.82</td><td>1.00</td><td>136.71</td></tr><tr><th>TP53</th><td>63</td><td>49</td><td>233</td><td>622</td><td>3.43</td><td>0.00000</td><td>0.00000</td><td>3.43</td><td>0.00000</td><td>0.00000</td><td>3.43</td><td>1.00</td><td>137.00</td></tr><tr><th>MDM2</th><td>10</td><td>102</td><td>38</td><td>817</td><td>2.11</td><td>0.06</td><td>8.12</td><td>2.11</td><td>0.04154</td><td>5.69056</td><td>2.11</td><td>0.98</td><td>134.71</td></tr><tr><th>NCOA3</th><td>8</td><td>104</td><td>15</td><td>840</td><td>4.31</td><td>0.00278</td><td>0.38135</td><td>4.31</td><td>0.00278</td><td>0.38135</td><td>4.31</td><td>1.00</td><td>136.92</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 13,
       "text": [
        "<IPython.core.display.HTML at 0x59ee2490>"
       ]
      }
     ],
     "prompt_number": 13
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which pathways are affected in MYC CN>4 but CN<6 samples"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "TP53_pathway=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\", \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\"]\n",
      "STAG1_pathway=[\"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\"]\n",
      "MLL_pathway=[\"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\"]\n",
      "BAP1_pathway=[\"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\"]\n",
      "KRAS_pathway=[\"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\"]\n",
      "HLA_pathway=[\"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\"]\n",
      "HER2_pathway=[\"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\"]\n",
      "CDON_pathway=[\"BOC\", \"CDON\"]\n",
      "KEAP1_pathway=[\"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\"]\n",
      "SPOP_pathway=[\"MYD88\", \"SPOP\"]\n",
      "RUNX1_pathway=[\"CBFB\", \"RUNX1\",\"ELF4\"]\n",
      "NOTCH_pathway=[\"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\"]\n",
      "ARID_pathway=[\"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\"]\n",
      "SMARC_pathway=[\"SMARCB1\", \"SMARCA4\"]\n",
      "SMC_pathway=[\"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\"]\n",
      "CLASP_pathway=[\"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "pathways={\"TP53_pathway\": TP53_pathway, \"STAG1_pathway\":STAG1_pathway, \"MLL_pathway\":MLL_pathway,\"BAP1_pathway\":BAP1_pathway,\"KRAS_pathway\":KRAS_pathway, \"HLA_pathway\":HLA_pathway,  \"HER2_pathway\":HER2_pathway, \"CDON_pathway\":CDON_pathway, \"KEAP1_pathway\":KEAP1_pathway,\"SPOP_pathway\":SPOP_pathway, \"RUNX1_pathway\":RUNX1_pathway, \"NOTCH_pathway\":NOTCH_pathway, \"ARID_pathway\":ARID_pathway, \"SMARC_pathway\":SMARC_pathway, \"SMC_pathway\":SMC_pathway,\"CLASP_pathway\":CLASP_pathway }\n",
      "\n",
      "MYC_pos_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_pos_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    copynumber = \"n/a\"\n",
      "    log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "    if (log2 != None):\n",
      "        copynumber = 2.0 * 2.0 ** log2\n",
      "    else:\n",
      "        copynumber=\"n/a\"\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if copynumber != \"n/a\" and pathway_affected and copynumber > 4 and copynumber < 6:\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and  copynumber > 4 and copynumber < 6:\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and pathway_affected and not  copynumber > 4:\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and not  copynumber > 4:\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAG1_pathway</th><td>0</td><td>93</td><td>55</td><td>800</td><td>0.00</td><td>0.00438</td><td>0.07014</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.00286</td><td>0.04578</td></tr><tr><th>KRAS_pathway</th><td>33</td><td>60</td><td>351</td><td>504</td><td>0.79</td><td>0.32</td><td>5.10</td><td>0.79</td><td>0.88</td><td>14.01</td><td>0.79</td><td>0.18</td><td>2.83</td></tr><tr><th>CLASP_pathway</th><td>4</td><td>89</td><td>21</td><td>834</td><td>1.78</td><td>0.30</td><td>4.75</td><td>1.78</td><td>0.22</td><td>3.58</td><td>1.78</td><td>0.91</td><td>14.57</td></tr><tr><th>SMARC_pathway</th><td>1</td><td>92</td><td>8</td><td>847</td><td>1.15</td><td>0.61</td><td>9.71</td><td>1.15</td><td>0.61</td><td>9.71</td><td>1.15</td><td>0.78</td><td>12.51</td></tr><tr><th>SPOP_pathway</th><td>0</td><td>93</td><td>5</td><td>850</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.60</td><td>9.54</td></tr><tr><th>RUNX1_pathway</th><td>2</td><td>91</td><td>57</td><td>798</td><td>0.31</td><td>0.11</td><td>1.77</td><td>0.31</td><td>0.99</td><td>15.77</td><td>0.31</td><td>0.06</td><td>0.91</td></tr><tr><th>MLL_pathway</th><td>14</td><td>79</td><td>113</td><td>742</td><td>1.16</td><td>0.63</td><td>10.09</td><td>1.16</td><td>0.36</td><td>5.74</td><td>1.16</td><td>0.75</td><td>11.99</td></tr><tr><th>TP53_pathway</th><td>75</td><td>18</td><td>502</td><td>353</td><td>2.93</td><td>0.00003</td><td>0.00044</td><td>2.93</td><td>0.00002</td><td>0.00025</td><td>2.93</td><td>1.00</td><td>16.00</td></tr><tr><th>ARID_pathway</th><td>6</td><td>87</td><td>61</td><td>794</td><td>0.90</td><td>1.00</td><td>16.00</td><td>0.90</td><td>0.66</td><td>10.58</td><td>0.90</td><td>0.51</td><td>8.12</td></tr><tr><th>KEAP1_pathway</th><td>4</td><td>89</td><td>29</td><td>826</td><td>1.28</td><td>0.56</td><td>8.90</td><td>1.28</td><td>0.41</td><td>6.54</td><td>1.28</td><td>0.78</td><td>12.55</td></tr><tr><th>SMC_pathway</th><td>5</td><td>88</td><td>24</td><td>831</td><td>1.97</td><td>0.19</td><td>3.11</td><td>1.97</td><td>0.15</td><td>2.34</td><td>1.97</td><td>0.94</td><td>15.10</td></tr><tr><th>NOTCH_pathway</th><td>17</td><td>76</td><td>123</td><td>732</td><td>1.33</td><td>0.35</td><td>5.67</td><td>1.33</td><td>0.19</td><td>3.12</td><td>1.33</td><td>0.88</td><td>14.00</td></tr><tr><th>BAP1_pathway</th><td>7</td><td>86</td><td>55</td><td>800</td><td>1.18</td><td>0.66</td><td>10.54</td><td>1.18</td><td>0.41</td><td>6.52</td><td>1.18</td><td>0.74</td><td>11.91</td></tr><tr><th>CDON_pathway</th><td>3</td><td>90</td><td>15</td><td>840</td><td>1.87</td><td>0.41</td><td>6.54</td><td>1.87</td><td>0.26</td><td>4.09</td><td>1.87</td><td>0.91</td><td>14.54</td></tr><tr><th>HLA_pathway</th><td>3</td><td>90</td><td>25</td><td>830</td><td>1.11</td><td>0.75</td><td>12.00</td><td>1.11</td><td>0.53</td><td>8.47</td><td>1.11</td><td>0.71</td><td>11.33</td></tr><tr><th>HER2_pathway</th><td>25</td><td>68</td><td>156</td><td>699</td><td>1.65</td><td>0.05</td><td>0.82</td><td>1.65</td><td>0.03394</td><td>0.54298</td><td>1.65</td><td>0.98</td><td>15.70</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 14,
       "text": [
        "<IPython.core.display.HTML at 0x528dc990>"
       ]
      }
     ],
     "prompt_number": 14
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which pathways are affected in MYC CN>6 samples"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "TP53_pathway=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\", \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\"]\n",
      "STAG1_pathway=[\"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\"]\n",
      "MLL_pathway=[\"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\"]\n",
      "BAP1_pathway=[\"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\"]\n",
      "KRAS_pathway=[\"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\"]\n",
      "HLA_pathway=[\"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\"]\n",
      "HER2_pathway=[\"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\"]\n",
      "CDON_pathway=[\"BOC\", \"CDON\"]\n",
      "KEAP1_pathway=[\"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\"]\n",
      "SPOP_pathway=[\"MYD88\", \"SPOP\"]\n",
      "RUNX1_pathway=[\"CBFB\", \"RUNX1\",\"ELF4\"]\n",
      "NOTCH_pathway=[\"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\"]\n",
      "ARID_pathway=[\"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\"]\n",
      "SMARC_pathway=[\"SMARCB1\", \"SMARCA4\"]\n",
      "SMC_pathway=[\"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\"]\n",
      "CLASP_pathway=[\"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "pathways={\"TP53_pathway\": TP53_pathway, \"STAG1_pathway\":STAG1_pathway, \"MLL_pathway\":MLL_pathway,\"BAP1_pathway\":BAP1_pathway,\"KRAS_pathway\":KRAS_pathway, \"HLA_pathway\":HLA_pathway,  \"HER2_pathway\":HER2_pathway, \"CDON_pathway\":CDON_pathway, \"KEAP1_pathway\":KEAP1_pathway,\"SPOP_pathway\":SPOP_pathway, \"RUNX1_pathway\":RUNX1_pathway, \"NOTCH_pathway\":NOTCH_pathway, \"ARID_pathway\":ARID_pathway, \"SMARC_pathway\":SMARC_pathway, \"SMC_pathway\":SMC_pathway,\"CLASP_pathway\":CLASP_pathway }\n",
      "\n",
      "MYC_pos_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_pos_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_affected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "MYC_neg_pathway_unaffected={\"TP53_pathway\": 0, \"STAG1_pathway\":0, \"MLL_pathway\":0,\"BAP1_pathway\":0,\"KRAS_pathway\":0, \"HLA_pathway\":0,  \"HER2_pathway\":0, \"CDON_pathway\":0, \"KEAP1_pathway\":0,\"SPOP_pathway\":0, \"RUNX1_pathway\":0, \"NOTCH_pathway\":0, \"ARID_pathway\":0, \"SMARC_pathway\":0, \"SMC_pathway\":0,\"CLASP_pathway\":0 }\n",
      "\n",
      "\n",
      "for sample in samples.values():\n",
      "    if not sample.somatic_mutation_data:\n",
      "        continue\n",
      "    copynumber = \"n/a\"\n",
      "    log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "    if (log2 != None):\n",
      "        copynumber = 2.0 * 2.0 ** log2\n",
      "    else:\n",
      "        copynumber=\"n/a\"\n",
      "    for pathway in pathways.keys():\n",
      "        pathway_affected = False\n",
      "        for gene in pathways[pathway]:\n",
      "            if gene in sample.genes_affected or sample.checkFocalGeneAmp(gene):\n",
      "                pathway_affected = True\n",
      "                break\n",
      "        if copynumber != \"n/a\" and pathway_affected and copynumber > 6:\n",
      "            MYC_pos_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and  copynumber > 6:\n",
      "            MYC_pos_pathway_unaffected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and pathway_affected and not  copynumber > 6:\n",
      "            MYC_neg_pathway_affected[pathway] += 1\n",
      "        elif copynumber != \"n/a\" and not pathway_affected and not  copynumber > 6:\n",
      "            MYC_neg_pathway_unaffected[pathway] += 1\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "for pathway in pathways.keys():\n",
      "   oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]])\n",
      "   oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='greater')\n",
      "   oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_pathway_affected[pathway], MYC_pos_pathway_unaffected[pathway]], [MYC_neg_pathway_affected[pathway], MYC_neg_pathway_unaffected[pathway]]], alternative='less')\n",
      "   s += \"<tr><th>\"+pathway+\"</th><td>\"+str(MYC_pos_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_pos_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_affected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+str(MYC_neg_pathway_unaffected[pathway])+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "   if pvalue < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "   if pvalue_greater < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   else:\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 16)+\"</td>\"\n",
      "   s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "   if pvalue_less < 0.05: \n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "   else: \n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "      s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 16)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Pathway</th><th>MYC + pathway affected</th><th>MYC + pathway unaffected</th><th>MYC - pathway affected</th><th>MYC - pathway unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>STAG1_pathway</th><td>1</td><td>18</td><td>55</td><td>893</td><td>0.90</td><td>1.00</td><td>16.00</td><td>0.90</td><td>0.68</td><td>10.91</td><td>0.90</td><td>0.70</td><td>11.16</td></tr><tr><th>KRAS_pathway</th><td>9</td><td>10</td><td>384</td><td>564</td><td>1.32</td><td>0.64</td><td>10.22</td><td>1.32</td><td>0.35</td><td>5.64</td><td>1.32</td><td>0.80</td><td>12.80</td></tr><tr><th>CLASP_pathway</th><td>0</td><td>19</td><td>25</td><td>923</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.61</td><td>9.68</td></tr><tr><th>SMARC_pathway</th><td>0</td><td>19</td><td>9</td><td>939</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.84</td><td>13.37</td></tr><tr><th>SPOP_pathway</th><td>0</td><td>19</td><td>5</td><td>943</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.91</td><td>14.49</td></tr><tr><th>RUNX1_pathway</th><td>0</td><td>19</td><td>59</td><td>889</td><td>0.00</td><td>0.62</td><td>9.98</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.30</td><td>4.78</td></tr><tr><th>MLL_pathway</th><td>2</td><td>17</td><td>127</td><td>821</td><td>0.76</td><td>1.00</td><td>16.00</td><td>0.76</td><td>0.74</td><td>11.91</td><td>0.76</td><td>0.52</td><td>8.39</td></tr><tr><th>TP53_pathway</th><td>15</td><td>4</td><td>577</td><td>371</td><td>2.41</td><td>0.15</td><td>2.44</td><td>2.41</td><td>0.08</td><td>1.33</td><td>2.41</td><td>0.97</td><td>15.55</td></tr><tr><th>ARID_pathway</th><td>2</td><td>17</td><td>67</td><td>881</td><td>1.55</td><td>0.64</td><td>10.24</td><td>1.55</td><td>0.40</td><td>6.37</td><td>1.55</td><td>0.85</td><td>13.63</td></tr><tr><th>KEAP1_pathway</th><td>1</td><td>18</td><td>33</td><td>915</td><td>1.54</td><td>0.50</td><td>7.95</td><td>1.54</td><td>0.50</td><td>7.95</td><td>1.54</td><td>0.86</td><td>13.74</td></tr><tr><th>SMC_pathway</th><td>0</td><td>19</td><td>29</td><td>919</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.56</td><td>8.92</td></tr><tr><th>NOTCH_pathway</th><td>2</td><td>17</td><td>140</td><td>808</td><td>0.68</td><td>1.00</td><td>16.00</td><td>0.68</td><td>0.79</td><td>12.70</td><td>0.68</td><td>0.46</td><td>7.28</td></tr><tr><th>BAP1_pathway</th><td>1</td><td>18</td><td>62</td><td>886</td><td>0.79</td><td>1.00</td><td>16.00</td><td>0.79</td><td>0.73</td><td>11.61</td><td>0.79</td><td>0.65</td><td>10.33</td></tr><tr><th>CDON_pathway</th><td>0</td><td>19</td><td>18</td><td>930</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.70</td><td>11.16</td></tr><tr><th>HLA_pathway</th><td>0</td><td>19</td><td>28</td><td>920</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>1.00</td><td>16.00</td><td>0.00</td><td>0.57</td><td>9.11</td></tr><tr><th>HER2_pathway</th><td>4</td><td>15</td><td>181</td><td>767</td><td>1.13</td><td>0.77</td><td>12.34</td><td>1.13</td><td>0.51</td><td>8.12</td><td>1.13</td><td>0.71</td><td>11.35</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 15,
       "text": [
        "<IPython.core.display.HTML at 0x59e98710>"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which genes of the Leiserson Pathway are affected at MYC copynumber > 4 but copynumber < 6 samples"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\",\n",
      "       \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\",\n",
      "       \"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\",\n",
      "       \"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\",\n",
      "       \"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\",\n",
      "       \"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\",\n",
      "       \"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\",\n",
      "       \"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\",\n",
      "       \"BOC\", \"CDON\",\n",
      "       \"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\",\n",
      "       \"MYD88\", \"SPOP\",\n",
      "       \"CBFB\", \"RUNX1\",\"ELF4\",\n",
      "       \"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\",\n",
      "       \"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\",\n",
      "       \"SMARCB1\", \"SMARCA4\",\n",
      "       \"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\",\n",
      "       \"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "gene_count = 0\n",
      "for gene in genes:\n",
      "    gene_count += 1\n",
      "    MYC_pos_gene_affected = 0\n",
      "    MYC_pos_gene_unaffected = 0\n",
      "    MYC_neg_gene_affected = 0\n",
      "    MYC_neg_gene_unaffected = 0\n",
      "\n",
      "    for sample in samples.values():\n",
      "        if not sample.somatic_mutation_data:\n",
      "            continue\n",
      "        copynumber = \"n/a\"\n",
      "        log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "        if (log2 != None):\n",
      "            copynumber = 2.0 * 2.0 ** log2\n",
      "        else:\n",
      "            copynumber=\"n/a\"\n",
      "        if sample.checkFocalGeneAmp(gene) or gene in sample.genes_affected:\n",
      "            if copynumber != \"n/a\" and copynumber > 4 and copynumber < 6:\n",
      "                MYC_pos_gene_affected += 1\n",
      "            elif  copynumber != \"n/a\" and not copynumber > 4:\n",
      "                MYC_neg_gene_affected += 1\n",
      "        else:\n",
      "            if copynumber != \"n/a\" and copynumber > 4 and copynumber < 6:\n",
      "                MYC_pos_gene_unaffected += 1\n",
      "            elif  copynumber != \"n/a\" and not copynumber > 4:\n",
      "                MYC_neg_gene_unaffected += 1\n",
      "\n",
      "    oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]])\n",
      "    oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='greater')\n",
      "    oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='less')\n",
      "\n",
      "    if (pvalue > 0.05 and pvalue_greater > 0.05 and pvalue_less > 0.05):\n",
      "        continue\n",
      "        \n",
      "    s += \"<tr><th>\"+gene+\"</th><td>\"+str(MYC_pos_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_pos_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "    if pvalue < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "    if pvalue_greater < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "    if pvalue_less < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "    else: \n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "print \"Genes: \"+str(gene_count)\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Genes: 134\n"
       ]
      },
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>TP53</th><td>52</td><td>41</td><td>233</td><td>622</td><td>3.39</td><td>0.00000</td><td>0.00001</td><td>3.39</td><td>0.00000</td><td>0.00001</td><td>3.39</td><td>1.00</td><td>134.00</td></tr><tr><th>CDKN2AIP</th><td>2</td><td>91</td><td>1</td><td>854</td><td>18.77</td><td>0.02676</td><td>3.58554</td><td>18.77</td><td>0.02676</td><td>3.58554</td><td>18.77</td><td>1.00</td><td>133.88</td></tr><tr><th>ZMIZ1</th><td>6</td><td>87</td><td>19</td><td>836</td><td>3.03</td><td>0.02885</td><td>3.86572</td><td>3.03</td><td>0.02885</td><td>3.86572</td><td>3.03</td><td>0.99</td><td>132.97</td></tr><tr><th>KRAS</th><td>7</td><td>86</td><td>26</td><td>829</td><td>2.60</td><td>0.03538</td><td>4.74100</td><td>2.60</td><td>0.03538</td><td>4.74100</td><td>2.60</td><td>0.99</td><td>132.50</td></tr><tr><th>PIK3CA</th><td>21</td><td>72</td><td>302</td><td>553</td><td>0.53</td><td>0.01521</td><td>2.03825</td><td>0.53</td><td>1.00</td><td>133.46</td><td>0.53</td><td>0.00810</td><td>1.08584</td></tr><tr><th>EGFR</th><td>7</td><td>86</td><td>17</td><td>838</td><td>4.01</td><td>0.00599</td><td>0.80333</td><td>4.01</td><td>0.00599</td><td>0.80333</td><td>4.01</td><td>1.00</td><td>133.83</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 16,
       "text": [
        "<IPython.core.display.HTML at 0x59eda9d0>"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Which genes of the Leiserson Pathway are affected at MYC copynumber > 6"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes=[\"CCND1\",\"CDKN2A\", \"CUL9\", \"NPM1\", \"PTEN\", \"TP53\",\"AGBL2\", \"ALS2CR8\", \"AMOTL1\", \"ANKRD12\", \"BRPF1\", \"CACHD1\", \"CARNS1\", \"CDKN2AIP\", \"CELSR3\", \"CHD8\", \"EPHA3\", \"HECW2\", \"IFT140\", \"IWS1\", \"MAGI2\", \"MAST3\", \"MDM4\", \"MLL5\", \"PLEKHA8\", \"PRDM2\", \"PRKRIR\", \"SCAPER\", \"SETD2\", \"SMG1\", \"SMG5\", \"SMG7\", \"SNRK\",\n",
      "       \"SPTBN2\", \"STK11IP\", \"STRADA\", \"SWAP70\", \"TEP1\", \"TRIP12\", \"TTLL5\", \"UACA\", \"ZFP91\", \"ZMIZ1\", \"ZNF227\", \"ZNF668\",\n",
      "       \"PDS5B\", \"STAG1\", \"STAG2\", \"WAPAL\",\"PDS5A\", \"RAD21\", \"SMC1A\",\n",
      "       \"KDM6A\", \"MLL2\", \"MLL3\",\"E2F3\", \"N4BP2\", \"PROSER1\",\n",
      "       \"ASXL1\", \"ASXL2\", \"BAP1\",\"ANKRD17\", \"FOXK1\", \"FOXK2\", \"KDM1B\",\n",
      "       \"BRAF\", \"KRAS\", \"PIK3CA\",\"ARHGAP4\", \"EIF2AK1\", \"FAM118B\", \"NRAS\", \"PLCE1\", \"RASGRF2\", \"RASGRP2\", \"RGL1\", \"VARS2\",\n",
      "       \"B2M\", \"CD1D\",\"HLA-A\", \"HLA-B\", \"P4HTM\",\n",
      "       \"EGFR\", \"ERBB2\",\"AREG\", \"ELF3\", \"ERBB4\", \"LRIG1\", \"OSMR\",\n",
      "       \"BOC\", \"CDON\",\n",
      "       \"KEAP1\", \"NFE2L2\",\"CHD6\", \"PTMA\", \"WAC\",\n",
      "       \"MYD88\", \"SPOP\",\n",
      "       \"CBFB\", \"RUNX1\",\"ELF4\",\n",
      "       \"FBXW7\", \"NOTCH1\",\"CCNE1\", \"DCLRE1C\", \"DLL1\", \"DTX4\", \"FAM101A\", \"FOXM1\", \"JAG1\", \"KLF5\", \"LFNG\", \"LPP\", \"MAML1\", \"MMS22L\", \"NOTCH3\", \"NOTCH4\", \"SHPRH\",\n",
      "       \"ARID1A\", \"PBRM1\",\"ADNP\", \"ARID1B\", \"ARID2\",\n",
      "       \"SMARCB1\", \"SMARCA4\",\n",
      "       \"NCAPD3\",\"SMC4\",\"NCAPG2\",\"NCAPH2\",\"SMC2\",\"NCAPD2\",\n",
      "       \"CLASP1\",\"CLASP2\",\"CLIP2\"]\n",
      "\n",
      "s = \"<table>\"\n",
      "s += \"<th></th><th colspan=\\\"4\\\">Sample</th><th colspan=\\\"3\\\">Fisher's exact test (two-sided)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided greater)</th><th colspan=\\\"3\\\">Fisher's exact test (one-sided less)</th></tr>\"\n",
      "s += \"<tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr>\"\n",
      "\n",
      "gene_count = 0\n",
      "for gene in genes:\n",
      "    gene_count += 1\n",
      "    MYC_pos_gene_affected = 0\n",
      "    MYC_pos_gene_unaffected = 0\n",
      "    MYC_neg_gene_affected = 0\n",
      "    MYC_neg_gene_unaffected = 0\n",
      "\n",
      "    for sample in samples.values():\n",
      "        if not sample.somatic_mutation_data:\n",
      "            continue\n",
      "        copynumber = \"n/a\"\n",
      "        log2 = sample.getLog2FromGene(gene_positions['MYC'].chrom,gene_positions['MYC'].start, gene_positions['MYC'].end)\n",
      "        if (log2 != None):\n",
      "            copynumber = 2.0 * 2.0 ** log2\n",
      "        else:\n",
      "            copynumber=\"n/a\"\n",
      "        if sample.checkFocalGeneAmp(gene) or gene in sample.genes_affected:\n",
      "            if copynumber != \"n/a\" and copynumber > 6:\n",
      "                MYC_pos_gene_affected += 1\n",
      "            elif  copynumber != \"n/a\" and not copynumber > 6:\n",
      "                MYC_neg_gene_affected += 1\n",
      "        else:\n",
      "            if copynumber != \"n/a\" and copynumber > 6:\n",
      "                MYC_pos_gene_unaffected += 1\n",
      "            elif  copynumber != \"n/a\" and not copynumber > 6:\n",
      "                MYC_neg_gene_unaffected += 1\n",
      "\n",
      "    oddsratio, pvalue = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]])\n",
      "    oddsratio_greater, pvalue_greater = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='greater')\n",
      "    oddsratio_less, pvalue_less = scipy.stats.fisher_exact([[MYC_pos_gene_affected, MYC_pos_gene_unaffected], [MYC_neg_gene_affected, MYC_neg_gene_unaffected]], alternative='less')\n",
      "\n",
      "    if (pvalue > 0.05 and pvalue_greater > 0.05 and pvalue_less > 0.05):\n",
      "        continue\n",
      "        \n",
      "    s += \"<tr><th>\"+gene+\"</th><td>\"+str(MYC_pos_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_pos_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_affected)+\"</td>\"\n",
      "    s += \"<td>\"+str(MYC_neg_gene_unaffected)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio)+\"</td>\"\n",
      "    if pvalue < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_greater)+\"</td>\"\n",
      "    if pvalue_greater < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    else:\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_greater * 134)+\"</td>\"\n",
      "    s += \"<td>\"+\"{:.2f}\".format(oddsratio_less)+\"</td>\"\n",
      "    if pvalue_less < 0.05: \n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.5f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "    else: \n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less)+\"</td>\"\n",
      "       s += \"<td>\"+\"{:.2f}\".format(pvalue_less * 134)+\"</td></tr>\"\n",
      "\n",
      "s += \"</table>\"\n",
      "print \"Genes: \"+str(gene_count)\n",
      "h = HTML(s);h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Genes: 134\n"
       ]
      },
      {
       "html": [
        "<table><th></th><th colspan=\"4\">Sample</th><th colspan=\"3\">Fisher's exact test (two-sided)</th><th colspan=\"3\">Fisher's exact test (one-sided greater)</th><th colspan=\"3\">Fisher's exact test (one-sided less)</th></tr><tr><th>Gene</th><th>MYC + Gene affected</th><th>MYC + Gene unaffected</th><th>MYC - Gene affected</th><th>MYC - Gene unaffected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th><th>Odd's ratio</th><th>p-value</th><th>p-value corrected</th></tr><tr><th>TP53</th><td>11</td><td>8</td><td>285</td><td>663</td><td>3.20</td><td>0.02022</td><td>2.70895</td><td>3.20</td><td>0.01164</td><td>1.56039</td><td>3.20</td><td>1.00</td><td>133.58</td></tr><tr><th>IFT140</th><td>2</td><td>17</td><td>7</td><td>941</td><td>15.82</td><td>0.01214</td><td>1.62627</td><td>15.82</td><td>0.01214</td><td>1.62627</td><td>15.82</td><td>1.00</td><td>133.93</td></tr><tr><th>MLL5</th><td>2</td><td>17</td><td>10</td><td>938</td><td>11.04</td><td>0.02148</td><td>2.87829</td><td>11.04</td><td>0.02148</td><td>2.87829</td><td>11.04</td><td>1.00</td><td>133.83</td></tr><tr><th>FOXK1</th><td>1</td><td>18</td><td>0</td><td>948</td><td>inf</td><td>0.01965</td><td>2.63289</td><td>inf</td><td>0.01965</td><td>2.63289</td><td>inf</td><td>1.00</td><td>134.00</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 17,
       "text": [
        "<IPython.core.display.HTML at 0x59e9a0d0>"
       ]
      }
     ],
     "prompt_number": 17
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": []
    }
   ],
   "metadata": {}
  }
 ]
}